%% Generated by Sphinx.
\def\sphinxdocclass{jupyterBook}
\documentclass[letterpaper,10pt,english]{jupyterBook}
\ifdefined\pdfpxdimen
   \let\sphinxpxdimen\pdfpxdimen\else\newdimen\sphinxpxdimen
\fi \sphinxpxdimen=.75bp\relax
\ifdefined\pdfimageresolution
    \pdfimageresolution= \numexpr \dimexpr1in\relax/\sphinxpxdimen\relax
\fi
%% let collapsible pdf bookmarks panel have high depth per default
\PassOptionsToPackage{bookmarksdepth=5}{hyperref}
%% turn off hyperref patch of \index as sphinx.xdy xindy module takes care of
%% suitable \hyperpage mark-up, working around hyperref-xindy incompatibility
\PassOptionsToPackage{hyperindex=false}{hyperref}
%% memoir class requires extra handling
\makeatletter\@ifclassloaded{memoir}
{\ifdefined\memhyperindexfalse\memhyperindexfalse\fi}{}\makeatother

\PassOptionsToPackage{warn}{textcomp}

\catcode`^^^^00a0\active\protected\def^^^^00a0{\leavevmode\nobreak\ }
\usepackage{cmap}
\usepackage{fontspec}
\defaultfontfeatures[\rmfamily,\sffamily,\ttfamily]{}
\usepackage{amsmath,amssymb,amstext}
\usepackage{polyglossia}
\setmainlanguage{english}



\setmainfont{FreeSerif}[
  Extension      = .otf,
  UprightFont    = *,
  ItalicFont     = *Italic,
  BoldFont       = *Bold,
  BoldItalicFont = *BoldItalic
]
\setsansfont{FreeSans}[
  Extension      = .otf,
  UprightFont    = *,
  ItalicFont     = *Oblique,
  BoldFont       = *Bold,
  BoldItalicFont = *BoldOblique,
]
\setmonofont{FreeMono}[
  Extension      = .otf,
  UprightFont    = *,
  ItalicFont     = *Oblique,
  BoldFont       = *Bold,
  BoldItalicFont = *BoldOblique,
]



\usepackage[Bjarne]{fncychap}
\usepackage[,numfigreset=1,mathnumfig]{sphinx}

\fvset{fontsize=\small}
\usepackage{geometry}


% Include hyperref last.
\usepackage{hyperref}
% Fix anchor placement for figures with captions.
\usepackage{hypcap}% it must be loaded after hyperref.
% Set up styles of URL: it should be placed after hyperref.
\urlstyle{same}


\usepackage{sphinxmessages}



        % Start of preamble defined in sphinx-jupyterbook-latex %
         \usepackage[Latin,Greek]{ucharclasses}
        \usepackage{unicode-math}
        % fixing title of the toc
        \addto\captionsenglish{\renewcommand{\contentsname}{Contents}}
        \hypersetup{
            pdfencoding=auto,
            psdextra
        }
        % End of preamble defined in sphinx-jupyterbook-latex %
        

\title{H-01C63}
\date{Jun 27, 2022}
\release{}
\author{Taeyoon kim}
\newcommand{\sphinxlogo}{\vbox{}}
\renewcommand{\releasename}{}
\makeindex
\begin{document}

\pagestyle{empty}
\sphinxmaketitle
\pagestyle{plain}
\sphinxtableofcontents
\pagestyle{normal}
\phantomsection\label{\detokenize{intro::doc}}


\begin{DUlineblock}{0em}
\item[] \sphinxstylestrong{\Large Introduction}
\end{DUlineblock}

\sphinxAtStartPar
The data in the report is intended to be a resource to guide the development and lead\sphinxhyphen{}optimization of a therapeutic protein. These data can be used in a preemptive fashion \sphinxhyphen{} for example, in the decision to substitute an exposed residue on the surface that may be prone to the kind of chemical modification that might affect the stability of the protein. They can also be used to assist the troubleshooting of problems that can arise in the course of an clinical development \sphinxhyphen{} for example if an therapeutic protein displays stability issues in storage, or unacceptably high levels of immunogenicity in early clinical trials.

\sphinxAtStartPar
There is always a great deal of risk involved in the development of any therapeutic molecule but experience has shown that the kind of data presented in this report is an invaluable tool for mitigating that risk \sphinxhyphen{} either by helping to identify potential problems before they occur, or by guiding the troubleshooting ofproblems that can occur during the antibody’s development and lead\sphinxhyphen{}optimization.

\begin{DUlineblock}{0em}
\item[] \sphinxstylestrong{\large About}
\end{DUlineblock}

\sphinxAtStartPar
H\sphinxhyphen{}01C73 inhibits the growth of cancer cells by targeting all classes of angiogenesis factors (VEGF) (VEGF\sphinxhyphen{}A, VEGF\sphinxhyphen{}B, Placental Growth Factor) that are overexpressed around cancer cells. H\sphinxhyphen{}01C65 is a material with high research difficulty due to its complex protein structure.

\sphinxAtStartPar
Panoros CEO Lim Hye\sphinxhyphen{}seong said, “H\sphinxhyphen{}01C65 is expected to have excellent efficacy as a treatment for cancer and VEGF\sphinxhyphen{}related diseases by itself, and the substance itself has already proven its value as a platform technology.”

\begin{DUlineblock}{0em}
\item[] \sphinxstylestrong{\large Protein sequence}
\end{DUlineblock}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{EFGRPFVEMYSEIPEIIHMTEGRELVIPCRVTSPNITVTLKKFPLDTLIPDGKRIIWDSRKGFIISNATYKEIGLLTCEATVNGHLYKTNYLTHRQTNTIIDVQISTPSPVELCVGETLVLNCTATTPANTRVQMTWSYPDEKNKNASVRRRIDQSNSHANIFYSVLTIDGMTNSDKGLYTCRVRSGPSFKSVNTSVHIYDCSNTGSGGEEKKKEKEKEEQEERSSDTPPTCPPCPAPELLGGPSVFLFPPKPKDTLMISRTPEVTCVVVDVSHEDPEVKFNWYVDGVEVHNAKTKPREEQYNSTYRVVSVLTVLHQDWLNGKEYKCKVSNKALPAPIEKTISKAKGQPREPQVYTLPPSRDELTKNQVSLTCLVKGFYPSDIAVEWESNGQPENNYKTTPPVLDSDGSFFLYSKLTVDKSRWQQGNVFSCSVMHEALHNHYTQKSLSLSPG}
\end{sphinxVerbatim}

\sphinxstepscope


\chapter{Protein parameters analysis}
\label{\detokenize{ipynb/chapter1:protein-parameters-analysis}}\label{\detokenize{ipynb/chapter1::doc}}
\sphinxAtStartPar
The program performs most of the same functions as the Expasy ProtParam tool.
\begin{itemize}
\item {} 
\sphinxAtStartPar
Molecular weight
\begin{itemize}
\item {} 
\sphinxAtStartPar
Amino acids are the building blocks that form polypeptides and ultimately proteins. Calculates the molecular weight of a protein.

\end{itemize}

\item {} 
\sphinxAtStartPar
Chemical composition
\begin{itemize}
\item {} 
\sphinxAtStartPar
A chemical formula is a way of presenting information about the chemical proportions of atoms that constitute a particular chemical compound or molecule, using chemical element symbols, numbers.

\end{itemize}

\item {} 
\sphinxAtStartPar
Extinction coefficient
\begin{itemize}
\item {} 
\sphinxAtStartPar
Extinction (or extinction coefficient) is defined as the ratio of maximum to minimum transmission of a beam of light that passes through a polarization optical train. extinction coefficient in units of  M\sphinxhyphen{}1 cm\sphinxhyphen{}1, at 280 nm measured in water.

\end{itemize}

\item {} 
\sphinxAtStartPar
Theoretical pI
\begin{itemize}
\item {} 
\sphinxAtStartPar
The isoelectric point (pI, pH(I), IEP), is the pH at which a molecule carries no net electrical charge or is electrically neutral in the statistical mean. The pI value can affect the solubility of a molecule at a given pH. Such molecules have minimum solubility in water or salt solutions at the pH that corresponds to their pI and often precipitate out of solution. Biological amphoteric molecules such as proteins contain both acidic and basic functional groups.

\end{itemize}

\item {} 
\sphinxAtStartPar
Aromaticity
\begin{itemize}
\item {} 
\sphinxAtStartPar
Calculate the aromaticity according to Lobry, 1994. Calculates the aromaticity value of a protein according to Lobry, 1994. It is simply the relative frequency of Phe+Trp+Tyr.

\end{itemize}

\item {} 
\sphinxAtStartPar
GRAVY
\begin{itemize}
\item {} 
\sphinxAtStartPar
The GRAVY value is calculated by adding the hydropathy value for each residue and dividing by the length of the sequence (Kyte and Doolittle; 1982). A higher value is more hydrophobic. A lower value is more hydrophilic.

\end{itemize}

\item {} 
\sphinxAtStartPar
Instability\_index
\begin{itemize}
\item {} 
\sphinxAtStartPar
Implementation of the method of Guruprasad et al. (1990, Protein Engineering, 4, 155\sphinxhyphen{}161). This method tests a protein for stability. Any value above 40 means the protein is unstable (=has a short half life).

\end{itemize}

\end{itemize}

\begin{sphinxuseclass}{cell}
\begin{sphinxuseclass}{tag_remove-input}\begin{sphinxVerbatimOutput}

\begin{sphinxuseclass}{cell_output}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZsh{} Name of target protein: \PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}H\PYGZus{}01C65
\PYGZsh{} Molecular weight(Dalton): \PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}50,818
\PYGZsh{} Total number of amino acid: \PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}452
\PYGZsh{} Chemical formula: \PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}C2248H3531N609O697S18
\PYGZsh{} Total number of atom is \PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}7103
\PYGZsh{} Extinction coefficient(reduced): \PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}58330
\PYGZsh{} Reduced Abs 0.1\PYGZpc{}(=1 g/L): \PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}1.148
\PYGZsh{} Extinction coefficient(non\PYGZhy{}reduced): \PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}59080
\PYGZsh{} Non\PYGZhy{}reduced Abs 0.1\PYGZpc{}(=1 g/L): \PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}1.163
\PYGZsh{} Theoretical pI: \PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}6.660
\PYGZsh{} Aromaticity: \PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}7.96\PYGZpc{}
\PYGZsh{} The GRAVY value is \PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}0.553
  H\PYGZus{}01C65 is more hydrophilic protein.
\PYGZsh{} The instability index: \PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}46.565
  H\PYGZus{}01C65 is seems unstable.
\end{sphinxVerbatim}

\end{sphinxuseclass}\end{sphinxVerbatimOutput}

\end{sphinxuseclass}
\end{sphinxuseclass}
\begin{sphinxuseclass}{cell}
\begin{sphinxuseclass}{tag_remove-input}\begin{sphinxVerbatimOutput}

\begin{sphinxuseclass}{cell_output}
\noindent\sphinxincludegraphics{{chapter1_2_0}.png}

\end{sphinxuseclass}\end{sphinxVerbatimOutput}

\end{sphinxuseclass}
\end{sphinxuseclass}

\section{Amino acid composition}
\label{\detokenize{ipynb/chapter1:amino-acid-composition}}
\sphinxAtStartPar
We can easily count the number of each type of amino acid.
\begin{itemize}
\item {} 
\sphinxAtStartPar
Number of each amino acids
\begin{itemize}
\item {} 
\sphinxAtStartPar
Simply counts the number times an amino acid is repeated in the protein sequence.

\end{itemize}

\end{itemize}

\begin{sphinxuseclass}{cell}
\begin{sphinxuseclass}{tag_remove-input}\begin{sphinxVerbatimOutput}

\begin{sphinxuseclass}{cell_output}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
Total number of positively charged residues(Arg + Lys):\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}52
Total number of negatively charged residues(Asp + Glu):\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}54
\end{sphinxVerbatim}

\noindent\sphinxincludegraphics{{chapter1_4_1}.png}

\end{sphinxuseclass}\end{sphinxVerbatimOutput}

\end{sphinxuseclass}
\end{sphinxuseclass}\begin{itemize}
\item {} 
\sphinxAtStartPar
Percent of amino acid contents

\end{itemize}

\begin{sphinxuseclass}{cell}
\begin{sphinxuseclass}{tag_remove-input}\begin{sphinxVerbatimOutput}

\begin{sphinxuseclass}{cell_output}
\noindent\sphinxincludegraphics{{chapter1_6_0}.png}

\end{sphinxuseclass}\end{sphinxVerbatimOutput}

\end{sphinxuseclass}
\end{sphinxuseclass}

\section{Potential sites of chemical modification}
\label{\detokenize{ipynb/chapter1:potential-sites-of-chemical-modification}}
\sphinxAtStartPar
An initial scan of the protein sequences is presented based purely upon sequence. If a structural analysis was also requested, this section should be used in conjunction with the molecular surface analysis described in a subsequent section. Any of the sites listed below could be candidates for further consideration if the molecular surface analysis shows that they are significantly exposed on the surface of the protein, increasing their propensity for chemical modification. The canonical sequence analysis is also helpful here, since each of these sites can also be considered in the context of their frequency of occurrence within the canonical library of homologous sequences.


\subsection{Potential deamidation positions}
\label{\detokenize{ipynb/chapter1:potential-deamidation-positions}}
\sphinxAtStartPar
Asparagine (N) and glutamine (Q) residues are particularly prone to deamidation when they are followed in the sequence by amino acids with smaller side chains, that leave the intervening peptide group more exposed. Deamidation proceeds much more quickly if the susceptible amino acid is followed by a small, flexible residue such as glycine whose low steric hindrance leaves the peptide group open for attack.
\begin{itemize}
\item {} 
\sphinxAtStartPar
Search patterns: ASN/GLN\sphinxhyphen{}ALA/GLY/SER/THR

\end{itemize}

\begin{sphinxuseclass}{cell}
\begin{sphinxuseclass}{tag_remove-input}\begin{sphinxVerbatimOutput}

\begin{sphinxuseclass}{cell_output}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
67\PYGZhy{}NA\PYGZhy{}68
83\PYGZhy{}NG\PYGZhy{}84
96\PYGZhy{}QT\PYGZhy{}97
98\PYGZhy{}NT\PYGZhy{}99
130\PYGZhy{}NT\PYGZhy{}131
146\PYGZhy{}NA\PYGZhy{}147
155\PYGZhy{}QS\PYGZhy{}156
157\PYGZhy{}NS\PYGZhy{}158
174\PYGZhy{}NS\PYGZhy{}175
194\PYGZhy{}NT\PYGZhy{}195
204\PYGZhy{}NT\PYGZhy{}205
292\PYGZhy{}NA\PYGZhy{}293
303\PYGZhy{}NS\PYGZhy{}304
321\PYGZhy{}NG\PYGZhy{}322
390\PYGZhy{}NG\PYGZhy{}391
425\PYGZhy{}QG\PYGZhy{}426
\end{sphinxVerbatim}

\end{sphinxuseclass}\end{sphinxVerbatimOutput}

\end{sphinxuseclass}
\end{sphinxuseclass}

\subsection{Potential o\sphinxhyphen{}linked glycosylation sites}
\label{\detokenize{ipynb/chapter1:potential-o-linked-glycosylation-sites}}
\sphinxAtStartPar
The O\sphinxhyphen{}linked glycosylation of serine and threonine residues seems to be particularly sensitive to the presence of one or more proline residues in their vicinity in the sequence, particularly in the 2\sphinxhyphen{}1 and +3 positions.
\begin{itemize}
\item {} 
\sphinxAtStartPar
Search patterns: PRO\sphinxhyphen{}SER/THR

\end{itemize}

\begin{sphinxuseclass}{cell}
\begin{sphinxuseclass}{tag_remove-input}\begin{sphinxVerbatimOutput}

\begin{sphinxuseclass}{cell_output}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
108\PYGZhy{}PS\PYGZhy{}109
188\PYGZhy{}PS\PYGZhy{}189
230\PYGZhy{}PT\PYGZhy{}231
244\PYGZhy{}PS\PYGZhy{}245
359\PYGZhy{}PS\PYGZhy{}360
380\PYGZhy{}PS\PYGZhy{}381
\end{sphinxVerbatim}

\end{sphinxuseclass}\end{sphinxVerbatimOutput}

\end{sphinxuseclass}
\end{sphinxuseclass}\begin{itemize}
\item {} 
\sphinxAtStartPar
Search patterns: SER/THR\sphinxhyphen{}X\sphinxhyphen{}X\sphinxhyphen{}PRO

\end{itemize}

\begin{sphinxuseclass}{cell}
\begin{sphinxuseclass}{tag_remove-input}\begin{sphinxVerbatimOutput}

\begin{sphinxuseclass}{cell_output}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
11\PYGZhy{}SEIP\PYGZhy{}14
47\PYGZhy{}TLIP\PYGZhy{}50
107\PYGZhy{}TPSP\PYGZhy{}110
226\PYGZhy{}SDTP\PYGZhy{}229
231\PYGZhy{}TCPP\PYGZhy{}234
260\PYGZhy{}SRTP\PYGZhy{}263
356\PYGZhy{}TLPP\PYGZhy{}359
399\PYGZhy{}TTPP\PYGZhy{}402
448\PYGZhy{}SLSP\PYGZhy{}451
\end{sphinxVerbatim}

\end{sphinxuseclass}\end{sphinxVerbatimOutput}

\end{sphinxuseclass}
\end{sphinxuseclass}

\subsection{Potential n\sphinxhyphen{}linked glycosylation sites}
\label{\detokenize{ipynb/chapter1:potential-n-linked-glycosylation-sites}}\begin{itemize}
\item {} 
\sphinxAtStartPar
Search patterns: ASN\sphinxhyphen{}X\sphinxhyphen{}SER/THR

\end{itemize}

\begin{sphinxuseclass}{cell}
\begin{sphinxuseclass}{tag_remove-input}\begin{sphinxVerbatimOutput}

\begin{sphinxuseclass}{cell_output}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
35\PYGZhy{}NIT\PYGZhy{}37
67\PYGZhy{}NAT\PYGZhy{}69
122\PYGZhy{}NCT\PYGZhy{}124
146\PYGZhy{}NAS\PYGZhy{}148
194\PYGZhy{}NTS\PYGZhy{}196
303\PYGZhy{}NST\PYGZhy{}305
\end{sphinxVerbatim}

\end{sphinxuseclass}\end{sphinxVerbatimOutput}

\end{sphinxuseclass}
\end{sphinxuseclass}

\section{Secondary structure fraction}
\label{\detokenize{ipynb/chapter1:secondary-structure-fraction}}
\sphinxAtStartPar
The fraction of amino acids that tend to be found in the three classical secondary structures. These are beta sheets, alpha helixes, and turns (where the residues change direction).
\begin{itemize}
\item {} 
\sphinxAtStartPar
Amino acids in helix: V, I, Y, F, W, L.

\item {} 
\sphinxAtStartPar
Amino acids in turn: N, P, G, S.

\item {} 
\sphinxAtStartPar
Amino acids in sheet: E, M, A, L.

\end{itemize}

\begin{sphinxuseclass}{cell}
\begin{sphinxuseclass}{tag_remove-input}\begin{sphinxVerbatimOutput}

\begin{sphinxuseclass}{cell_output}
\noindent\sphinxincludegraphics{{chapter1_16_0}.png}

\end{sphinxuseclass}\end{sphinxVerbatimOutput}

\end{sphinxuseclass}
\end{sphinxuseclass}

\section{Secondary structure prediction}
\label{\detokenize{ipynb/chapter1:secondary-structure-prediction}}
\sphinxAtStartPar
Protein secondary structure prediction is one of the most important and challenging problems in bioinformatics. Here in, the P\sphinxhyphen{}SEA algorithm that to predict the secondary structures of proteins sequences based only on knowledge of their primary structure.

\begin{sphinxuseclass}{cell}
\begin{sphinxuseclass}{tag_remove-input}\begin{sphinxVerbatimOutput}

\begin{sphinxuseclass}{cell_output}
\noindent\sphinxincludegraphics{{chapter1_18_0}.png}

\end{sphinxuseclass}\end{sphinxVerbatimOutput}

\end{sphinxuseclass}
\end{sphinxuseclass}

\section{Structural analysis}
\label{\detokenize{ipynb/chapter1:structural-analysis}}

\subsection{Detection of disulfide bonds}
\label{\detokenize{ipynb/chapter1:detection-of-disulfide-bonds}}
\sphinxAtStartPar
This function detects disulfide bridges in protein structures. Then the detected disulfide bonds are visualized and added to the bonds attribute of the AtomArray. The employed criteria for disulfide bonds are quite simple in this case: the atoms of two cystein residues must be in a vicinity of Å and the dihedral angle of must be .

\begin{sphinxuseclass}{cell}
\begin{sphinxuseclass}{tag_remove-input}\begin{sphinxVerbatimOutput}

\begin{sphinxuseclass}{cell_output}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
    B     361  CYS SG     S        \PYGZhy{}8.239    2.300   19.435
    B     419  CYS SG     S        \PYGZhy{}6.772    1.269   20.460
\end{sphinxVerbatim}

\noindent\sphinxincludegraphics{{chapter1_20_1}.png}

\end{sphinxuseclass}\end{sphinxVerbatimOutput}

\end{sphinxuseclass}
\end{sphinxuseclass}
\sphinxAtStartPar
The found disulfide bonds are visualized with the help of Matplotlib: The amino acid sequence is written on the X\sphinxhyphen{}axis and the disulfide bonds are depicted by yellow semi\sphinxhyphen{}ellipses.


\subsection{Calculation of protein diameter}
\label{\detokenize{ipynb/chapter1:calculation-of-protein-diameter}}
\sphinxAtStartPar
This calculates the diameter of a protein defined as the maximum pairwise atom distance.

\begin{sphinxuseclass}{cell}
\begin{sphinxuseclass}{tag_remove-input}\begin{sphinxVerbatimOutput}

\begin{sphinxuseclass}{cell_output}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZsh{} Diameter of H\PYGZus{}01C65 is: \PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}116.214 Angstrong.
\end{sphinxVerbatim}

\end{sphinxuseclass}\end{sphinxVerbatimOutput}

\end{sphinxuseclass}
\end{sphinxuseclass}

\section{Protein Scales}
\label{\detokenize{ipynb/chapter1:protein-scales}}
\sphinxAtStartPar
Protein scales are a way of measuring certain attributes of residues over the length of the peptide sequence using a sliding window. Scales are comprised of values for each amino acid based on different physical and chemical properties, such as hydrophobicity, secondary structure tendencies, and surface accessibility. As opposed to some chain\sphinxhyphen{}level measures like overall molecule behavior, scales allow a more granular understanding of how smaller sections of the sequence will behave.
\begin{itemize}
\item {} 
\sphinxAtStartPar
kd → Kyte \& Doolittle Index of Hydrophobicity

\item {} 
\sphinxAtStartPar
Flex → Normalized average flexibility parameters (B\sphinxhyphen{}values)

\item {} 
\sphinxAtStartPar
hw → Hopp \& Wood Index of Hydrophilicity

\item {} 
\sphinxAtStartPar
em → Emini Surface fractional probability (Surface Accessibility)

\end{itemize}


\subsection{Hydrophobicity index}
\label{\detokenize{ipynb/chapter1:hydrophobicity-index}}
\sphinxAtStartPar
hydrophobicity is the physical property of a molecule that is seemingly repelled from a mass of water (known as a hydrophobe).

\begin{sphinxuseclass}{cell}
\begin{sphinxuseclass}{tag_remove-input}\begin{sphinxVerbatimOutput}

\begin{sphinxuseclass}{cell_output}
\noindent\sphinxincludegraphics{{chapter1_24_0}.png}

\end{sphinxuseclass}\end{sphinxVerbatimOutput}

\end{sphinxuseclass}
\end{sphinxuseclass}

\subsection{Hydrophilicity index}
\label{\detokenize{ipynb/chapter1:hydrophilicity-index}}
\sphinxAtStartPar
Hydrophilicity is the tendency of a molecule to be solvated by water.

\begin{sphinxuseclass}{cell}
\begin{sphinxuseclass}{tag_remove-input}\begin{sphinxVerbatimOutput}

\begin{sphinxuseclass}{cell_output}
\noindent\sphinxincludegraphics{{chapter1_26_0}.png}

\end{sphinxuseclass}\end{sphinxVerbatimOutput}

\end{sphinxuseclass}
\end{sphinxuseclass}

\subsection{Flexibility index}
\label{\detokenize{ipynb/chapter1:flexibility-index}}
\sphinxAtStartPar
Proteins are dynamic entities, and they possess an inherent flexibility that allows them to function through molecular interactions within the cell.

\begin{sphinxuseclass}{cell}
\begin{sphinxuseclass}{tag_remove-input}\begin{sphinxVerbatimOutput}

\begin{sphinxuseclass}{cell_output}
\noindent\sphinxincludegraphics{{chapter1_28_0}.png}

\end{sphinxuseclass}\end{sphinxVerbatimOutput}

\end{sphinxuseclass}
\end{sphinxuseclass}

\subsection{Surface accessibility}
\label{\detokenize{ipynb/chapter1:surface-accessibility}}
\sphinxAtStartPar
Data describing the solvent\sphinxhyphen{}accessible surface of a molecule is of great utility in the development of that molecule as a therapeutic, particularly in the case of antibodies. In the context of this report, the most obvious application of molecular surface data is in combination with the potential sites of chemical modification, described in the previous section. Proteins are known to undergo many different chemical modifications as a result of interactions with their aqueous environment. The probability and kinetic rate of such a modification is greatly enhanced by the degree of exposure of the potential modification site to the solvent environment. The solvent\sphinxhyphen{}accessible surface for each residue depends upon the degree of exposure of the residue on the surface, but also on the size of the residue side chain.

\begin{sphinxuseclass}{cell}
\begin{sphinxuseclass}{tag_remove-input}\begin{sphinxVerbatimOutput}

\begin{sphinxuseclass}{cell_output}
\noindent\sphinxincludegraphics{{chapter1_30_0}.png}

\end{sphinxuseclass}\end{sphinxVerbatimOutput}

\end{sphinxuseclass}
\end{sphinxuseclass}

\subsection{Instability index}
\label{\detokenize{ipynb/chapter1:instability-index}}
\sphinxAtStartPar
The instability index provides an estimate of the stability of your protein in a test tube. Statistical analysis of 12 unstable and 32 stable proteins has revealed that there are certain dipeptides, the occurence of which is significantly different in the unstable proteins compared with those in the stable ones.

\begin{sphinxuseclass}{cell}
\begin{sphinxuseclass}{tag_remove-input}\begin{sphinxVerbatimOutput}

\begin{sphinxuseclass}{cell_output}
\noindent\sphinxincludegraphics{{chapter1_32_0}.png}

\end{sphinxuseclass}\end{sphinxVerbatimOutput}

\end{sphinxuseclass}
\end{sphinxuseclass}
\sphinxstepscope


\chapter{Immunogenicity analysis}
\label{\detokenize{ipynb/chapter2:immunogenicity-analysis}}\label{\detokenize{ipynb/chapter2::doc}}
\sphinxAtStartPar
We uses the method of removing and/or reducing potential T\sphinxhyphen{}cell epitopes, as an approach to the management of the immunogenicity of biologics. The protein sequence is scanned in silico, for sequences that have a strong binding signature for a family of 50 MHC Class II receptors , whose alleles cover 96 – 98\% of the human population. The presented histograms for each variable region sequence, show the average (for the n positively\sphinxhyphen{}testing MHC II alleles) of epitope strength at each position as a percentage for all epitopes above a threshold of 20\%. At each position in the sequence, the number of alleles scoring above the threshold is shown above the histogram at that position. The epitopes of most concern for the antibody’s immunogenicity are therefore those that have not just the highest average score per allele (as shown by the histogram), but which also score above the threshold across more alleles, since these epitopes are more likely to engender an immune response in a larger fraction of the patient population.

\sphinxAtStartPar
Experience using in silico algorithms of this kind in conjunction with laboratory immunogenicity assays has shown that epitopes below this threshold do not generally contribute significantly to the protein’s immunogenicity. The number of alleles, the affected alleles and their individual scores are also listed in the detailed analyses below each histogram figure.

\sphinxAtStartPar
The raw immunogenicity score quoted is the total over all epitopes above the threshold for all affected alleles. The normalized immunogenicity score is this raw score divided by the sequence length, and represents epitope strength per unit sequence to enable comparisons of protein sequences of different lengths.


\section{MHC class 1}
\label{\detokenize{ipynb/chapter2:mhc-class-1}}
\sphinxAtStartPar
Class I major histocompatibility complex (MHC) molecules bind, and present to T cells, short peptides derived from intracellular processing of proteins. The peptide repertoire of a specific molecule is to a large extent determined by the molecular structure accommodating so\sphinxhyphen{}called main anchor positions of the presented peptide.

\sphinxAtStartPar
Their function is to display peptide fragments of proteins from within the cell to cytotoxic T cells; this will trigger an immediate response from the immune system against a particular non\sphinxhyphen{}self antigen displayed with the help of an MHC class I protein. Because MHC class I molecules present peptides derived from cytosolic proteins, the pathway of MHC class I presentation is often called cytosolic or endogenous pathway.%
\begin{footnote}[1]\sphinxAtStartFootnote
Kimball’s Biology Pages, Histocompatibility Molecules
%
\end{footnote}
\begin{itemize}
\item {} 
\sphinxAtStartPar
MHC class 1 superset
\begin{itemize}
\item {} 
\sphinxAtStartPar
HLA\sphinxhyphen{}A01:01, HLA\sphinxhyphen{}A02:01, HLA\sphinxhyphen{}A03:01, HLA\sphinxhyphen{}A24:02, HLA\sphinxhyphen{}B07:02, HLA\sphinxhyphen{}B40:01

\end{itemize}

\end{itemize}


\subsection{Predicts binding of peptides to MHC class1}
\label{\detokenize{ipynb/chapter2:predicts-binding-of-peptides-to-mhc-class1}}
\begin{sphinxuseclass}{cell}
\begin{sphinxuseclass}{tag_remove-input}
\begin{sphinxuseclass}{tag_remove-output}
\end{sphinxuseclass}
\end{sphinxuseclass}
\end{sphinxuseclass}
\begin{sphinxuseclass}{cell}
\begin{sphinxuseclass}{tag_remove-input}\begin{sphinxVerbatimOutput}

\begin{sphinxuseclass}{cell_output}
\noindent\sphinxincludegraphics{{chapter2_2_0}.png}

\end{sphinxuseclass}\end{sphinxVerbatimOutput}

\end{sphinxuseclass}
\end{sphinxuseclass}

\subsection{Top10 strong binding peptide}
\label{\detokenize{ipynb/chapter2:top10-strong-binding-peptide}}
\begin{sphinxuseclass}{cell}
\begin{sphinxuseclass}{tag_remove-input}\begin{sphinxVerbatimOutput}

\begin{sphinxuseclass}{cell_output}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
           allele  peptide
Core                      
QPREPQVTL       3        3
DSDGSFFLY       2        2
TPANTRVQM       2        2
\PYGZhy{}SDGSFFLY       1        1
SPN\PYGZhy{}ITVTL       1        1
MYIPEIIHM       1        1
MYSEIPEII       1        1
NYKTTPPVL       1        1
QPREPQVYT       1        1
QSNSHNIFY       1        1
\end{sphinxVerbatim}

\end{sphinxuseclass}\end{sphinxVerbatimOutput}

\end{sphinxuseclass}
\end{sphinxuseclass}

\subsection{Frequency of binding peptide}
\label{\detokenize{ipynb/chapter2:frequency-of-binding-peptide}}
\begin{sphinxuseclass}{cell}
\begin{sphinxuseclass}{tag_remove-input}\begin{sphinxVerbatimOutput}

\begin{sphinxuseclass}{cell_output}
\noindent\sphinxincludegraphics{{chapter2_6_0}.png}

\end{sphinxuseclass}\end{sphinxVerbatimOutput}

\end{sphinxuseclass}
\end{sphinxuseclass}

\section{MHC class 2}
\label{\detokenize{ipynb/chapter2:mhc-class-2}}
\sphinxAtStartPar
MHC Class II molecules are a class of major histocompatibility complex (MHC) molecules normally found only on professional antigen\sphinxhyphen{}presenting cells such as dendritic cells, mononuclear phagocytes, some endothelial cells, thymic epithelial cells, and B cells. These cells are important in initiating immune responses.
\begin{itemize}
\item {} 
\sphinxAtStartPar
MHC class 2 allele superset
\begin{itemize}
\item {} 
\sphinxAtStartPar
DRB1\_0101,DRB1\_0102,DRB1\_0103,DRB1\_0104,DRB1\_0105,DRB1\_0106,DRB1\_0107,DRB1\_0108,DRB1\_0109,DRB1\_0110

\end{itemize}

\end{itemize}


\subsection{Predicts binding of peptides to MHC class2}
\label{\detokenize{ipynb/chapter2:predicts-binding-of-peptides-to-mhc-class2}}
\begin{sphinxuseclass}{cell}
\begin{sphinxuseclass}{tag_remove-input}
\begin{sphinxuseclass}{tag_remove-output}
\end{sphinxuseclass}
\end{sphinxuseclass}
\end{sphinxuseclass}
\begin{sphinxuseclass}{cell}
\begin{sphinxuseclass}{tag_remove-input}\begin{sphinxVerbatimOutput}

\begin{sphinxuseclass}{cell_output}
\noindent\sphinxincludegraphics{{chapter2_9_0}.png}

\end{sphinxuseclass}\end{sphinxVerbatimOutput}

\end{sphinxuseclass}
\end{sphinxuseclass}

\subsection{Top10 binding peptide}
\label{\detokenize{ipynb/chapter2:top10-binding-peptide}}
\begin{sphinxuseclass}{cell}
\begin{sphinxuseclass}{tag_remove-input}\begin{sphinxVerbatimOutput}

\begin{sphinxuseclass}{cell_output}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
           allele
Core             
IHMTEGREL      31
YKCKVSNKA      30
YRVVSVLTV      11
IDVQISTPS       7
ISKAKGQPR       7
YKTTPPVLD       6
FIISNATYK       4
FVEMYSEIP       4
WDSRKGFII       4
VQISTPSPV       3
\end{sphinxVerbatim}

\end{sphinxuseclass}\end{sphinxVerbatimOutput}

\end{sphinxuseclass}
\end{sphinxuseclass}

\subsection{Frequency of binding peptide}
\label{\detokenize{ipynb/chapter2:id2}}
\begin{sphinxuseclass}{cell}
\begin{sphinxuseclass}{tag_remove-input}\begin{sphinxVerbatimOutput}

\begin{sphinxuseclass}{cell_output}
\noindent\sphinxincludegraphics{{chapter2_13_0}.png}

\end{sphinxuseclass}\end{sphinxVerbatimOutput}

\end{sphinxuseclass}
\end{sphinxuseclass}

\section{Appendix}
\label{\detokenize{ipynb/chapter2:appendix}}
\sphinxAtStartPar
nothing yet.

\begin{sphinxuseclass}{cell}
\begin{sphinxuseclass}{tag_remove-input}
\begin{sphinxuseclass}{tag_remove-output}
\end{sphinxuseclass}
\end{sphinxuseclass}
\end{sphinxuseclass}
\begin{sphinxuseclass}{cell}
\begin{sphinxuseclass}{tag_remove-input}
\begin{sphinxuseclass}{tag_remove-output}
\end{sphinxuseclass}
\end{sphinxuseclass}
\end{sphinxuseclass}
\begin{sphinxuseclass}{cell}
\begin{sphinxuseclass}{tag_remove-input}
\begin{sphinxuseclass}{tag_remove-output}
\end{sphinxuseclass}
\end{sphinxuseclass}
\end{sphinxuseclass}
\begin{sphinxuseclass}{cell}
\begin{sphinxuseclass}{tag_remove-input}
\begin{sphinxuseclass}{tag_remove-output}
\end{sphinxuseclass}
\end{sphinxuseclass}
\end{sphinxuseclass}

\bigskip\hrule\bigskip








\renewcommand{\indexname}{Index}
\printindex
\end{document}