{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a8207d39",
   "metadata": {},
   "outputs": [],
   "source": [
    "import biotite.sequence as seq\n",
    "import biotite.sequence.align as align\n",
    "import numpy as np\n",
    "\n",
    "alph = seq.ProteinSequence.alphabet\n",
    "# Load the standard protein substitution matrix, which is BLOSUM62\n",
    "matrix = align.SubstitutionMatrix.std_protein_matrix()\n",
    "print(\"\\nBLOSUM62\\n\")\n",
    "print(matrix)\n",
    "# Load another matrix from internal database\n",
    "matrix = align.SubstitutionMatrix(alph, alph, \"BLOSUM50\")\n",
    "# Load a matrix dictionary representation,\n",
    "# modify it, and create the SubstitutionMatrix\n",
    "# (The dictionary could be alternatively loaded from a string containing\n",
    "# the matrix in NCBI format)\n",
    "matrix_dict = align.SubstitutionMatrix.dict_from_db(\"BLOSUM62\")\n",
    "matrix_dict[(\"P\",\"Y\")] = 100\n",
    "matrix = align.SubstitutionMatrix(alph, alph, matrix_dict)\n",
    "# And now create a matrix by directly provding the ndarray\n",
    "# containing the similarity scores\n",
    "# (identity matrix in our case)\n",
    "scores = np.identity(len(alph), dtype=int)\n",
    "matrix = align.SubstitutionMatrix(alph, alph, scores)\n",
    "print(\"\\n\\nIdentity matrix\\n\")\n",
    "print(matrix)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d66d9f99",
   "metadata": {},
   "outputs": [],
   "source": [
    "seq1 = seq.ProteinSequence(\"BIQTITE\")\n",
    "seq2 = seq.ProteinSequence(\"IQLITE\")\n",
    "matrix = align.SubstitutionMatrix.std_protein_matrix()\n",
    "print(\"\\nLocal alignment\")\n",
    "alignments = align.align_optimal(seq1, seq2, matrix, local=True)\n",
    "for ali in alignments:\n",
    "    print(ali)\n",
    "print(\"Global alignment\")\n",
    "alignments = align.align_optimal(seq1, seq2, matrix, local=False)\n",
    "for ali in alignments:\n",
    "    print(ali)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8eafc27e",
   "metadata": {},
   "outputs": [],
   "source": [
    "import matplotlib.pyplot as plt\n",
    "import biotite.sequence.graphics as graphics\n",
    "\n",
    "fig, ax = plt.subplots(figsize=(2.0, 0.8))\n",
    "graphics.plot_alignment_similarity_based(\n",
    "    ax, alignments[0], matrix=matrix, symbols_per_line=len(alignments[0])\n",
    ")\n",
    "fig.tight_layout()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7734b4af",
   "metadata": {},
   "outputs": [],
   "source": [
    "import matplotlib.pyplot as plt\n",
    "import biotite.sequence as seq\n",
    "import biotite.sequence.align as align\n",
    "import biotite.sequence.io.fasta as fasta\n",
    "import biotite.database.entrez as entrez\n",
    "import biotite.sequence.graphics as graphics\n",
    "\n",
    "# Download and parse protein sequences of avidin and streptavidin\n",
    "fasta_file = fasta.FastaFile.read(entrez.fetch_single_file(\n",
    "    [\"CAC34569\", \"ACL82594\"], None, \"protein\", \"fasta\"\n",
    "))\n",
    "for name, sequence in fasta_file.items():\n",
    "    if \"CAC34569\" in name:\n",
    "        avidin_seq = seq.ProteinSequence(sequence)\n",
    "    elif \"ACL82594\" in name:\n",
    "        streptavidin_seq = seq.ProteinSequence(sequence)\n",
    "\n",
    "# Get BLOSUM62 matrix\n",
    "matrix = align.SubstitutionMatrix.std_protein_matrix()\n",
    "# Perform pairwise sequence alignment with affine gap penalty\n",
    "# Terminal gaps are not penalized\n",
    "alignments = align.align_optimal(avidin_seq, streptavidin_seq, matrix,\n",
    "                                 gap_penalty=(-10, -1), terminal_penalty=False)\n",
    "\n",
    "# Draw first and only alignment\n",
    "# The color intensity indicates the similiarity\n",
    "fig = plt.figure(figsize=(8.0, 2.5))\n",
    "ax = fig.add_subplot(111)\n",
    "graphics.plot_alignment_similarity_based(\n",
    "    ax, alignments[0], matrix=matrix, labels=[\"Avidin\", \"Streptavidin\"],\n",
    "    show_numbers=True, show_line_position=True\n",
    ")\n",
    "fig.tight_layout()\n",
    "\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "054ca5ba",
   "metadata": {},
   "source": [
    "# Calculation of codon usage\n",
    "\n",
    "This script creates a table for codon usage in the Escherichia coli K-12 strain. Codon usage describes the frequencies of the codons that code for an amino acid. These frequencies are expected to reflect the abundance of the respective tRNAs in the target organism. The codon usage differs from species to species. Hence, it is important to look at the codon usage of the organism of interest for applications like codon optimization etc.\n",
    "\n",
    "For the computation of the codon usage we will have a look into the annotated Escherichia coli K-12 genome: The script extracts all coding sequences (CDS) from the genome and counts the total number of each codon in these sequences. Then the relative frequencies are calculated by dividing the total number of occurrences of each codon by the total number of occurrences of the respective amino acid. In order to improve the performance, the script mostly works with symbol codes (see tutorial) instead of the symbols itself.\n",
    "\n",
    "At first we fetch the genome from the NCBI Entrez database (Accession: U00096) as GenBank file and parse it into an AnnotatedSequence. Then we create a dictionary that will store the total codon frequencies later on. As already mentioned, the script works with symbol codes. Consequently, each codon in the dictionary is described as 3 integers instead of 3 letters."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "52ba0084",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Code source: Patrick Kunzmann\n",
    "# License: BSD 3 clause\n",
    "\n",
    "import tempfile\n",
    "import itertools\n",
    "import numpy as np\n",
    "import biotite.sequence as seq\n",
    "import biotite.sequence.io.genbank as gb\n",
    "import biotite.sequence.io.fasta as fasta\n",
    "import biotite.database.entrez as entrez\n",
    "\n",
    "\n",
    "# Get the E. coli K-12 genome as annotated sequence\n",
    "gb_file = gb.GenBankFile.read(\n",
    "    entrez.fetch(\"U00096\", tempfile.gettempdir(), \"gb\", \"nuccore\", \"gb\")\n",
    ")\n",
    "# We are only interested in CDS features\n",
    "k12_genome = gb.get_annotated_sequence(gb_file, include_only=[\"CDS\"])\n",
    "\n",
    "\n",
    "# This dictionary will count how often each codon occurs in the genome\n",
    "# For increased performance the dictionary uses symbol codes ([0 3 2])\n",
    "# instead of symbols (['A' 'T' 'G']) as keys\n",
    "codon_counter = {\n",
    "    codon: 0 for codon\n",
    "    in itertools.product( *([range(len(k12_genome.sequence.alphabet))] * 3) )\n",
    "}\n",
    "# For demonstration purposes print the 64 codons in symbol code form\n",
    "# print(list(codon_counter.keys()))\n",
    "\n",
    "# Iterate over all CDS features\n",
    "for cds in k12_genome.annotation:\n",
    "    # Obtain the sequence of the CDS\n",
    "    cds_seq = k12_genome[cds]\n",
    "    if len(cds_seq) % 3 != 0:\n",
    "        # A CDS' length should be a multiple of 3,\n",
    "        # otherwise the CDS is malformed\n",
    "        continue\n",
    "    # Iterate over the sequence in non-overlapping frames of 3\n",
    "    # and count the occurence of each codon\n",
    "    for i in range(0, len(cds_seq), 3):\n",
    "        codon_code = tuple(cds_seq.code[i:i+3])\n",
    "        codon_counter[codon_code] += 1\n",
    "\n",
    "# Convert the total frequencies into relative frequencies\n",
    "# for each amino acid\n",
    "# The NCBI codon table with ID 11 is the bacterial codon table\n",
    "table = seq.CodonTable.load(11)\n",
    "# As the script uses symbol codes, each amino acid is represented by a\n",
    "# number between 0 and 19, instead of the single letter code\n",
    "for amino_acid_code in range(20):\n",
    "    # Find all codons coding for the amino acid\n",
    "    # The codons are also in symbol code format, e.g. ATG -> (0, 3, 2)\n",
    "    codon_codes_for_aa = table[amino_acid_code]\n",
    "    # Get the total amount of codon occurrences for the amino acid\n",
    "    total = 0\n",
    "    for codon_code in codon_codes_for_aa:\n",
    "        total += codon_counter[codon_code]\n",
    "    # Replace the total frequencies with relative frequencies\n",
    "    # and print it\n",
    "    for codon_code in codon_codes_for_aa:\n",
    "        # Convert total frequencies into relative frequencies\n",
    "        codon_counter[codon_code] /= total\n",
    "        # The rest of this code block prints the codon usage table\n",
    "        # in human readable format\n",
    "        amino_acid = seq.ProteinSequence.alphabet.decode(amino_acid_code)\n",
    "        # Convert the symbol codes for each codon into symbols...\n",
    "        # ([0,3,2] -> ['A' 'T' 'G'])\n",
    "        codon = k12_genome.sequence.alphabet.decode_multiple(codon_code)\n",
    "        # ...and represent as string\n",
    "        # (['A' 'T' 'G'] -> \"ATG\")\n",
    "        codon = \"\".join(codon)\n",
    "        freq = codon_counter[codon_code]\n",
    "        print(f\"{amino_acid}   {codon}   {freq:.2f}\")\n",
    "    print()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2e398f87",
   "metadata": {},
   "source": [
    "As expected the dictionary encodes each codon as tuple of 3 numbers, where 0 represents A, 1 C, 2 G and 3 T. These mappings are defined by the alphabet of the genomic sequence.\n",
    "\n",
    "In the next step the occurrences of codons in the coding sequences are counted, the relative frequencies are calculated and the codon table is printed.\n",
    "\n",
    "The codon usage table can be used to optimize recombinant protein expression by designing the DNA sequence for the target protein according to the codon usage. This is called codon optimization. However, there is a variety of different algorithms for codon optimization. For simplicity reasons, this example uses an approach, where for every amino acid always the most frequently occuring codon is used.\n",
    "\n",
    "In the follwing we will derive a codon optimized DNA sequence of streptavidin for expression in E. coli K-12.\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5a232506",
   "metadata": {},
   "outputs": [],
   "source": [
    "# For each amino acid, select the codon with maximum codon usage\n",
    "# Again, symbol codes are used here\n",
    "opt_codons = {}\n",
    "for amino_acid_code in range(20):\n",
    "    codon_codes_for_aa = table[amino_acid_code]\n",
    "    # Find codon with maximum frequency\n",
    "    max_freq = 0\n",
    "    best_codon_code = None\n",
    "    for codon_code in codon_codes_for_aa:\n",
    "        if codon_counter[codon_code] > max_freq:\n",
    "            max_freq = codon_counter[codon_code]\n",
    "            best_codon_code = codon_code\n",
    "    # Map the amino acid to the codon with maximum frequency\n",
    "    opt_codons[amino_acid_code] = best_codon_code\n",
    "\n",
    "# Fetch the streptavidin protein sequence from Streptomyces avidinii\n",
    "fasta_file = fasta.FastaFile.read(\n",
    "    entrez.fetch(\"P22629\", None, \"fasta\", \"protein\", \"fasta\")\n",
    ")\n",
    "strep_prot_seq = fasta.get_sequence(fasta_file)\n",
    "# Create a DNA sequence from the protein sequence\n",
    "# using the optimal codons\n",
    "strep_dna_seq = seq.NucleotideSequence()\n",
    "strep_dna_seq.code = np.concatenate(\n",
    "    [opt_codons[amino_acid_code] for amino_acid_code in strep_prot_seq.code]\n",
    ")\n",
    "# Add stop codon\n",
    "strep_dna_seq += seq.NucleotideSequence(\"TAA\")\n",
    "# Put the DNA sequence into a FASTA file\n",
    "fasta_file = fasta.FastaFile()\n",
    "fasta_file[\"Codon optimized sequence\"] = str(strep_dna_seq)\n",
    "# Print the contents of the created FASTA file\n",
    "print(fasta_file)\n",
    "# In a real application it would be written onto the hard drive via\n",
    "# fasta_file.write(\"../data/some_file.fasta\")"
   ]
  }
 ],
 "metadata": {
  "celltoolbar": "Edit Metadata",
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.4"
  },
  "tags": [
   "hide-input"
  ]
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
