{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Canonical sequence analysis\n",
    "\n",
    "Canonical sequence analysis (CSA) is a useful method for identifying potential trouble spots in\n",
    "an antibody being developed for clinical use and thereby mitigating some of the potential\n",
    "upstream development risk for that antibody. This analysis considers the relative frequency of\n",
    "occurrence of the amino acid at each position in the scanned sequence, against a library of\n",
    "homologous antibody sequences. The CSA makes no assumptions about which organism the\n",
    "antibody sequences are derived from, nor which antibody frameworks were included in these\n",
    "sequences - particularly since it is common for antibodies to be derived from different organisms\n",
    "or different antibody frameworks from the same organism’s germline. Furthermore, many\n",
    "antibody sequences destined for clinical use, have been edited during lead optimization, and/or\n",
    "have been fully or partially humanized. The CSA simply compares the antibody sequences with\n",
    "the closest homologs identified using a global BLAST alignment, and compiles the frequencies1\n",
    "of occurrence of the amino acids at each position as an indicator of how representative or\n",
    "“typical” each amino acid is at a given position in the canonical antibody sequence represented\n",
    "by the alignment.\n",
    "\n",
    "Plotting these frequencies against the sequence facilitates the rapid identification of sequence\n",
    "positions that contain amino acids that are atypical at those positions. For the sequence\n",
    "positions in and around the complementarity-determining regions (CDRs) where recombination\n",
    "and hypermutation events produce more novel sequences, it is expected that these regions will\n",
    "be far less canonical and more unique, in keeping with the specificity of the antibody for its\n",
    "antigen.\n",
    "\n",
    "The low frequency positions in the sequence do not necessarily entail potential trouble spots in\n",
    "the antibody but they can identify the positions in the sequence with the highest potential risk\n",
    "that may merit further investigation. This is because antibodies generated by the immune\n",
    "system undergo a significant degree of natural selection, which is a significant filter for\n",
    "eliminating non-functional sequences.\n",
    "\n",
    "Closer examination of these atypical positions may reveal for example, functionally conservative\n",
    "substitutions that are unlikely to be problematic, or conversely - very unusual substitutions that\n",
    "may turn out, upon further investigation, to be essential, neutral or even detrimental to the\n",
    "antibody’s folding and/or function. A canonical sequence analysis is also a useful tool for any\n",
    "subsequent engineering of the antibody – to reduce its potential immunogenicity for example.\n",
    "CSA can provide invaluable guidance to the protein engineer in the selection of substitutions at\n",
    "any given position that will have a higher probability of preserving the structure and function of\n",
    "the antibody (which is always the most challenging aspect of protein engineering).\n",
    "\n",
    "The CSA is also very useful in highlighting the potential sites of chemical modification that are\n",
    "identified in a subsequent section. A site identified as potentially susceptible to a chemical\n",
    "modification that impairs the antibody’s stability and/or function, is less likely to be problematic if\n",
    "it is a residue that is often observed at that position in homologous antibodies. A highly unusual\n",
    "residue at that position, however, might merit further consideration for substitution. In the case\n",
    "that this report includes a structural analysis, data from the molecular surface analysis described\n",
    "later on, can also be used in conjunction with the CSA, to determine whether the potential\n",
    "modification site is both non-canonical and exposed on the antibody surface. Both of these\n",
    "factors would make it a higher risk for chemical modification.\n",
    "\n",
    "Presented below, are the CSA histograms for the heavy and light chain variable regions of your\n",
    "antibody. In each graph, the sequence itself is shown on the x-axis, with the percentage\n",
    "frequency in the library of the amino acid at each position, shown on the y-axis. The library was\n",
    "generated by a BLAST search of each antibody sequence against the database of homologous\n",
    "protein sequences, with the aligned set of the most closely homologous sequences being used\n",
    "for compiling the statistics.\n",
    "\n",
    "At each position in the sequence, the degree to which the amino acid is ‘typical’ at that position\n",
    "is represented by a green histogram showing its frequency of occurrence at the correspondingly\n",
    "aligned positions of the homologous antibody sequences in the library. Regions of the sequence\n",
    "that are more unique to the current antibody appear as gaps or shorter histograms. For the\n",
    "purposes of this analysis, we consider an amino acid at a given position in the sequence whose\n",
    "frequency in the library falls below about 10%, as being atypical.\n",
    "\n",
    "Along with the canonical sequence analysis plots, pie charts for the light and heavy chains are\n",
    "presented, that express the fractions (as percentages) of the aligned sequences derived from\n",
    "the different organisms represented in the alignments. For clarity, only the five most commonly\n",
    "observed organisms in the alignments, are shown in each pie chart."
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.10"
  },
  "widgets": {
   "application/vnd.jupyter.widget-state+json": {
    "state": {},
    "version_major": 2,
    "version_minor": 0
   }
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}