
<!DOCTYPE html>

<html>
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>Potential sites of chemical modification &#8212; Afilbercept</title>
    
  <link href="../_static/css/theme.css" rel="stylesheet">
  <link href="../_static/css/index.ff1ffe594081f20da1ef19478df9384b.css" rel="stylesheet">

    
  <link rel="stylesheet"
    href="../_static/vendor/fontawesome/5.13.0/css/all.min.css">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../_static/vendor/fontawesome/5.13.0/webfonts/fa-solid-900.woff2">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../_static/vendor/fontawesome/5.13.0/webfonts/fa-brands-400.woff2">

    
      

    
    <link rel="stylesheet" type="text/css" href="../_static/pygments.css" />
    <link rel="stylesheet" type="text/css" href="../_static/sphinx-book-theme.css?digest=c3fdc42140077d1ad13ad2f1588a4309" />
    <link rel="stylesheet" type="text/css" href="../_static/togglebutton.css" />
    <link rel="stylesheet" type="text/css" href="../_static/copybutton.css" />
    <link rel="stylesheet" type="text/css" href="../_static/mystnb.css" />
    <link rel="stylesheet" type="text/css" href="../_static/sphinx-thebe.css" />
    <link rel="stylesheet" type="text/css" href="../_static/panels-main.c949a650a448cc0ae9fd3441c0e17fb0.css" />
    <link rel="stylesheet" type="text/css" href="../_static/panels-variables.06eb56fa6e07937060861dad626602ad.css" />
    
  <link rel="preload" as="script" href="../_static/js/index.be7d3bbb2ef33a8344ce.js">

    <script data-url_root="../" id="documentation_options" src="../_static/documentation_options.js"></script>
    <script src="../_static/jquery.js"></script>
    <script src="../_static/underscore.js"></script>
    <script src="../_static/doctools.js"></script>
    <script src="../_static/clipboard.min.js"></script>
    <script src="../_static/copybutton.js"></script>
    <script>let toggleHintShow = 'Click to show';</script>
    <script>let toggleHintHide = 'Click to hide';</script>
    <script>let toggleOpenOnPrint = 'true';</script>
    <script src="../_static/togglebutton.js"></script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown, .tag_hide_input div.cell_input, .tag_hide-input div.cell_input, .tag_hide_output div.cell_output, .tag_hide-output div.cell_output, .tag_hide_cell.cell, .tag_hide-cell.cell';</script>
    <script src="../_static/sphinx-book-theme.d59cb220de22ca1c485ebbdc042f0030.js"></script>
    <script>const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"
const thebe_selector = ".thebe,.cell"
const thebe_selector_input = "pre"
const thebe_selector_output = ".output, .cell_output"
</script>
    <script async="async" src="../_static/sphinx-thebe.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="Detection of disulfide bonds" href="chapter3.html" />
    <link rel="prev" title="Protein parameters analysis" href="chapter1.html" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <meta name="docsearch:language" content="None">
    

    <!-- Google Analytics -->
    
  </head>
  <body data-spy="scroll" data-target="#bd-toc-nav" data-offset="80">
    
    <div class="container-fluid" id="banner"></div>

    

    <div class="container-xl">
      <div class="row">
          
<div class="col-12 col-md-3 bd-sidebar site-navigation show" id="site-navigation">
    
        <div class="navbar-brand-box">
    <a class="navbar-brand text-wrap" href="../index.html">
      
        <!-- `logo` is deprecated in Sphinx 4.0, so remove this when we stop supporting 3 -->
        
      
      
      <img src="../_static/logo.png" class="logo" alt="logo">
      
      
      <h1 class="site-logo" id="site-title">Afilbercept</h1>
      
    </a>
</div><form class="bd-search d-flex align-items-center" action="../search.html" method="get">
  <i class="icon fas fa-search"></i>
  <input type="search" class="form-control" name="q" id="search-input" placeholder="Search this book..." aria-label="Search this book..." autocomplete="off" >
</form><nav class="bd-links" id="bd-docs-nav" aria-label="Main">
    <div class="bd-toc-item active">
        <ul class="nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="../intro.html">
   About this report
  </a>
 </li>
</ul>
<ul class="current nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="chapter1.html">
   Protein parameters analysis
  </a>
 </li>
 <li class="toctree-l1 current active">
  <a class="current reference internal" href="#">
   Potential sites of chemical modification
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="chapter3.html">
   Detection of disulfide bonds
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="chapter4.html">
   Immunogenicity analysis
  </a>
 </li>
</ul>

    </div>
</nav> <!-- To handle the deprecated key -->

<div class="navbar_extra_footer">
  Powered by <a href="https://jupyterbook.org">Jupyter Book</a>
</div>

</div>


          


          
<main class="col py-md-3 pl-md-4 bd-content overflow-auto" role="main">
    
    <div class="topbar container-xl fixed-top">
    <div class="topbar-contents row">
        <div class="col-12 col-md-3 bd-topbar-whitespace site-navigation show"></div>
        <div class="col pl-md-4 topbar-main">
            
            <button id="navbar-toggler" class="navbar-toggler ml-0" type="button" data-toggle="collapse"
                data-toggle="tooltip" data-placement="bottom" data-target=".site-navigation" aria-controls="navbar-menu"
                aria-expanded="true" aria-label="Toggle navigation" aria-controls="site-navigation"
                title="Toggle navigation" data-toggle="tooltip" data-placement="left">
                <i class="fas fa-bars"></i>
                <i class="fas fa-arrow-left"></i>
                <i class="fas fa-arrow-up"></i>
            </button>
            
            
<div class="dropdown-buttons-trigger">
    <button id="dropdown-buttons-trigger" class="btn btn-secondary topbarbtn" aria-label="Download this page"><i
            class="fas fa-download"></i></button>

    <div class="dropdown-buttons">
        <!-- ipynb file if we had a myst markdown file -->
        
        <!-- Download raw file -->
        <a class="dropdown-buttons" href="../_sources/ipynb/chapter2.ipynb"><button type="button"
                class="btn btn-secondary topbarbtn" title="Download source file" data-toggle="tooltip"
                data-placement="left">.ipynb</button></a>
        <!-- Download PDF via print -->
        <button type="button" id="download-print" class="btn btn-secondary topbarbtn" title="Print to PDF"
                onclick="printPdf(this)" data-toggle="tooltip" data-placement="left">.pdf</button>
    </div>
</div>

            <!-- Source interaction buttons -->

<div class="dropdown-buttons-trigger">
    <button id="dropdown-buttons-trigger" class="btn btn-secondary topbarbtn"
        aria-label="Connect with source repository"><i class="fab fa-github"></i></button>
    <div class="dropdown-buttons sourcebuttons">
        <a class="repository-button"
            href="https://github.com/partrita/ProDevAnalysis"><button type="button" class="btn btn-secondary topbarbtn"
                data-toggle="tooltip" data-placement="left" title="Source repository"><i
                    class="fab fa-github"></i>repository</button></a>
        <a class="issues-button"
            href="https://github.com/partrita/ProDevAnalysis/issues/new?title=Issue%20on%20page%20%2Fipynb/chapter2.html&body=Your%20issue%20content%20here."><button
                type="button" class="btn btn-secondary topbarbtn" data-toggle="tooltip" data-placement="left"
                title="Open an issue"><i class="fas fa-lightbulb"></i>open issue</button></a>
        
    </div>
</div>

            <!-- Full screen (wrap in <a> to have style consistency -->

<a class="full-screen-button"><button type="button" class="btn btn-secondary topbarbtn" data-toggle="tooltip"
        data-placement="bottom" onclick="toggleFullScreen()" aria-label="Fullscreen mode"
        title="Fullscreen mode"><i
            class="fas fa-expand"></i></button></a>

            <!-- Launch buttons -->

<div class="dropdown-buttons-trigger">
    <button id="dropdown-buttons-trigger" class="btn btn-secondary topbarbtn"
        aria-label="Launch interactive content"><i class="fas fa-rocket"></i></button>
    <div class="dropdown-buttons">
        
        <a class="binder-button" href="https://mybinder.org/v2/gh/partrita/ProDevAnalysis/master?urlpath=tree/demo-report/ipynb/chapter2.ipynb"><button type="button"
                class="btn btn-secondary topbarbtn" title="Launch Binder" data-toggle="tooltip"
                data-placement="left"><img class="binder-button-logo"
                    src="../_static/images/logo_binder.svg"
                    alt="Interact on binder">Binder</button></a>
        
        
        
        
    </div>
</div>

        </div>

        <!-- Table of contents -->
        <div class="d-none d-md-block col-md-2 bd-toc show noprint">
            
            <div class="tocsection onthispage pt-5 pb-3">
                <i class="fas fa-list"></i> Contents
            </div>
            <nav id="bd-toc-nav" aria-label="Page">
                <ul class="visible nav section-nav flex-column">
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#potential-deamidation-positions">
   Potential deamidation positions
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#search-patterns-asn-gln-ala-gly-ser-thr">
     Search patterns: ASN/GLN-ALA/GLY/SER/THR
    </a>
   </li>
  </ul>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#potential-o-linked-glycosylation-sites">
   Potential o-linked glycosylation sites
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#search-patterns-pro-ser-thr">
     Search patterns: PRO-SER/THR
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#search-patterns-ser-thr-x-x-pro">
     Search patterns: SER/THR-X-X-PRO
    </a>
   </li>
  </ul>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#potential-n-linked-glycosylation-sites">
   Potential n-linked glycosylation sites
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#search-patterns-asn-x-ser-thr">
     Search patterns: ASN-X-SER/THR
    </a>
   </li>
  </ul>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#secondary-structure-fraction">
   Secondary structure fraction
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#secondary-structure-prediction">
   Secondary structure prediction
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#protein-scales">
   Protein Scales
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#hydrophobicity-index">
     Hydrophobicity index
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#hydrophilicity-index">
     Hydrophilicity index
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#flexibility-index">
     Flexibility index
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#surface-accessibility">
     Surface accessibility
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#instability-index">
     Instability index
    </a>
   </li>
  </ul>
 </li>
</ul>

            </nav>
        </div>
    </div>
</div>
    <div id="main-content" class="row">
        <div class="col-12 col-md-9 pl-md-3 pr-md-0">
            <!-- Table of contents that is only displayed when printing the page -->
            <div id="jb-print-docs-body" class="onlyprint">
                <h1>Potential sites of chemical modification</h1>
                <!-- Table of contents -->
                <div id="print-main-content">
                    <div id="jb-print-toc">
                        
                        <div>
                            <h2> Contents </h2>
                        </div>
                        <nav aria-label="Page">
                            <ul class="visible nav section-nav flex-column">
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#potential-deamidation-positions">
   Potential deamidation positions
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#search-patterns-asn-gln-ala-gly-ser-thr">
     Search patterns: ASN/GLN-ALA/GLY/SER/THR
    </a>
   </li>
  </ul>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#potential-o-linked-glycosylation-sites">
   Potential o-linked glycosylation sites
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#search-patterns-pro-ser-thr">
     Search patterns: PRO-SER/THR
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#search-patterns-ser-thr-x-x-pro">
     Search patterns: SER/THR-X-X-PRO
    </a>
   </li>
  </ul>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#potential-n-linked-glycosylation-sites">
   Potential n-linked glycosylation sites
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#search-patterns-asn-x-ser-thr">
     Search patterns: ASN-X-SER/THR
    </a>
   </li>
  </ul>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#secondary-structure-fraction">
   Secondary structure fraction
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#secondary-structure-prediction">
   Secondary structure prediction
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#protein-scales">
   Protein Scales
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#hydrophobicity-index">
     Hydrophobicity index
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#hydrophilicity-index">
     Hydrophilicity index
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#flexibility-index">
     Flexibility index
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#surface-accessibility">
     Surface accessibility
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#instability-index">
     Instability index
    </a>
   </li>
  </ul>
 </li>
</ul>

                        </nav>
                    </div>
                </div>
            </div>
            
              <div>
                
  <div class="tex2jax_ignore mathjax_ignore section" id="potential-sites-of-chemical-modification">
<h1>Potential sites of chemical modification<a class="headerlink" href="#potential-sites-of-chemical-modification" title="Permalink to this headline">¶</a></h1>
<p>An initial scan of the antibody sequences is presented based purely upon sequence. If a structural analysis was also requested, this section should be used in conjunction with the molecular surface analysis described in a subsequent section. Any of the sites listed below could be candidates for further consideration if the molecular surface analysis shows that they are significantly exposed on the surface of the antibody, increasing their propensity for chemical modification. The canonical sequence analysis is also helpful here, since each of these sites can also be considered in the context of their frequency of occurrence within the canonical library of homologous sequences.</p>
<p>This is an example of a footnote.<a class="footnote-reference brackets" href="#footnote1" id="id1">1</a></p>
<div class="cell tag_remove-input docutils container">
</div>
<div class="section" id="potential-deamidation-positions">
<h2>Potential deamidation positions<a class="headerlink" href="#potential-deamidation-positions" title="Permalink to this headline">¶</a></h2>
<p>Asparagine (N) and glutamine (Q) residues are particularly prone to deamidation when they are followed in the sequence by amino acids with smaller side chains, that leave the intervening peptide group more exposed. Deamidation proceeds much more quickly if the susceptible amino acid is followed by a small, flexible residue such as glycine whose low steric hindrance leaves the peptide group open for attack.</p>
<div class="section" id="search-patterns-asn-gln-ala-gly-ser-thr">
<h3>Search patterns: ASN/GLN-ALA/GLY/SER/THR<a class="headerlink" href="#search-patterns-asn-gln-ala-gly-ser-thr" title="Permalink to this headline">¶</a></h3>
<div class="cell tag_remove-input docutils container">
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>68-NA-69
84-NG-85
97-QT-98
99-NT-100
158-QS-159
180-QG-181
196-NS-197
271-NA-272
282-NS-283
300-NG-301
369-NG-370
404-QG-405
</pre></div>
</div>
</div>
</div>
</div>
</div>
<div class="section" id="potential-o-linked-glycosylation-sites">
<h2>Potential o-linked glycosylation sites<a class="headerlink" href="#potential-o-linked-glycosylation-sites" title="Permalink to this headline">¶</a></h2>
<p>The O-linked glycosylation of serine and threonine residues seems to be particularly sensitive to
the presence of one or more proline residues in their vicinity in the sequence, particularly in the2
-1 and +3 positions.</p>
<div class="section" id="search-patterns-pro-ser-thr">
<h3>Search patterns: PRO-SER/THR<a class="headerlink" href="#search-patterns-pro-ser-thr" title="Permalink to this headline">¶</a></h3>
<div class="cell tag_remove-input docutils container">
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>108-PS-109
141-PS-142
223-PS-224
338-PS-339
359-PS-360
</pre></div>
</div>
</div>
</div>
</div>
<div class="section" id="search-patterns-ser-thr-x-x-pro">
<h3>Search patterns: SER/THR-X-X-PRO<a class="headerlink" href="#search-patterns-ser-thr-x-x-pro" title="Permalink to this headline">¶</a></h3>
<div class="cell tag_remove-input docutils container">
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>3-TGRP-6
12-SEIP-15
48-TLIP-51
210-TCPP-213
239-SRTP-242
335-TLPP-338
378-TTPP-381
427-SLSP-430
</pre></div>
</div>
</div>
</div>
</div>
</div>
<div class="section" id="potential-n-linked-glycosylation-sites">
<h2>Potential n-linked glycosylation sites<a class="headerlink" href="#potential-n-linked-glycosylation-sites" title="Permalink to this headline">¶</a></h2>
<div class="section" id="search-patterns-asn-x-ser-thr">
<h3>Search patterns: ASN-X-SER/THR<a class="headerlink" href="#search-patterns-asn-x-ser-thr" title="Permalink to this headline">¶</a></h3>
<div class="cell tag_remove-input docutils container">
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>36-NIT-38
68-NAT-70
123-NCT-125
196-NST-198
282-NST-284
</pre></div>
</div>
</div>
</div>
</div>
</div>
<div class="section" id="secondary-structure-fraction">
<h2>Secondary structure fraction<a class="headerlink" href="#secondary-structure-fraction" title="Permalink to this headline">¶</a></h2>
<p>A very useful method — .secondary_structure_fraction() — returns the fraction of amino acids that tend to be found in the three classical secondary structures. These are beta sheets, alpha helixes, and turns (where the residues change direction).</p>
<p>Secondary_structure_fraction: This methods returns a list of the fraction of amino acids which tend to be in helix, turn or sheet.</p>
<ul class="simple">
<li><p>Amino acids in helix: V, I, Y, F, W, L.</p></li>
<li><p>Amino acids in turn: N, P, G, S.</p></li>
<li><p>Amino acids in sheet: E, M, A, L.</p></li>
</ul>
<div class="cell tag_remove-input docutils container">
<div class="cell_output docutils container">
<img alt="../_images/chapter2_11_0.png" src="../_images/chapter2_11_0.png" />
</div>
</div>
</div>
<div class="section" id="secondary-structure-prediction">
<h2>Secondary structure prediction<a class="headerlink" href="#secondary-structure-prediction" title="Permalink to this headline">¶</a></h2>
<p>Protein secondary structure prediction is one of the most important and challenging problems in bioinformatics. Here in, the P-SEA algorithm that to predict the secondary structures of proteins sequences based only on knowledge of their primary structure.</p>
<div class="cell tag_remove-input docutils container">
<div class="cell_output docutils container">
<img alt="../_images/chapter2_13_0.png" src="../_images/chapter2_13_0.png" />
</div>
</div>
</div>
<div class="section" id="protein-scales">
<h2>Protein Scales<a class="headerlink" href="#protein-scales" title="Permalink to this headline">¶</a></h2>
<p>Protein scales are a way of measuring certain attributes of residues over the length of the peptide sequence using a sliding window. Scales are comprised of values for each amino acid based on different physical and chemical properties, such as hydrophobicity, secondary structure tendencies, and surface accessibility. As opposed to some chain-level measures like overall molecule behavior, scales allow a more granular understanding of how smaller sections of the sequence will behave.</p>
<p>Some common scales include:</p>
<div class="highlight-none notranslate"><div class="highlight"><pre><span></span>- kd → Kyte &amp; Doolittle Index of Hydrophobicity
- Flex → Normalized average flexibility parameters (B-values)
- hw → Hopp &amp; Wood Index of Hydrophilicity
- em → Emini Surface fractional probability (Surface Accessibility)
</pre></div>
</div>
<div class="section" id="hydrophobicity-index">
<h3>Hydrophobicity index<a class="headerlink" href="#hydrophobicity-index" title="Permalink to this headline">¶</a></h3>
<p>hydrophobicity is the physical property of a molecule that is seemingly repelled from a mass of water (known as a hydrophobe).</p>
<div class="cell tag_remove-input docutils container">
<div class="cell_output docutils container">
<img alt="../_images/chapter2_15_0.png" src="../_images/chapter2_15_0.png" />
</div>
</div>
</div>
<div class="section" id="hydrophilicity-index">
<h3>Hydrophilicity index<a class="headerlink" href="#hydrophilicity-index" title="Permalink to this headline">¶</a></h3>
<p>Hydrophilicity is the tendency of a molecule to be solvated by water.</p>
<div class="cell tag_hide-input docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">plt</span><span class="o">.</span><span class="n">figure</span><span class="p">(</span><span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">12</span><span class="p">,</span><span class="mi">6</span><span class="p">))</span>
<span class="n">plt</span><span class="o">.</span><span class="n">style</span><span class="o">.</span><span class="n">use</span><span class="p">(</span><span class="s1">&#39;ggplot&#39;</span><span class="p">)</span>
<span class="n">_list</span> <span class="o">=</span> <span class="p">[</span><span class="s1">&#39;hw&#39;</span><span class="p">]</span>

<span class="k">for</span> <span class="n">i</span> <span class="ow">in</span> <span class="n">_list</span><span class="p">:</span>
    <span class="n">method_to_call</span> <span class="o">=</span> <span class="nb">getattr</span><span class="p">(</span><span class="n">ProtParamData</span><span class="p">,</span> <span class="n">i</span><span class="p">)</span>
    <span class="n">data</span> <span class="o">=</span> <span class="n">analyzed_seq</span><span class="o">.</span><span class="n">protein_scale</span><span class="p">(</span><span class="n">window</span><span class="o">=</span><span class="mi">7</span><span class="p">,</span> <span class="n">param_dict</span><span class="o">=</span><span class="n">method_to_call</span><span class="p">)</span>
    <span class="n">plt</span><span class="o">.</span><span class="n">plot</span><span class="p">(</span><span class="n">data</span><span class="p">,</span> <span class="n">label</span><span class="o">=</span><span class="n">i</span><span class="p">,</span> <span class="n">alpha</span><span class="o">=</span><span class="mf">0.7</span><span class="p">)</span>

<span class="n">plt</span><span class="o">.</span><span class="n">title</span><span class="p">(</span><span class="s1">&#39;Hydrophilicity&#39;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">xlabel</span><span class="p">(</span><span class="s1">&#39;Position&#39;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">ylabel</span><span class="p">(</span><span class="s1">&#39;&#39;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">legend</span><span class="p">()</span>
<span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="../_images/chapter2_17_0.png" src="../_images/chapter2_17_0.png" />
</div>
</div>
</div>
<div class="section" id="flexibility-index">
<h3>Flexibility index<a class="headerlink" href="#flexibility-index" title="Permalink to this headline">¶</a></h3>
<p>Proteins are dynamic entities, and they possess an inherent flexibility that allows them to function through molecular interactions within the cell.</p>
<div class="cell tag_remove-input docutils container">
<div class="cell_output docutils container">
<img alt="../_images/chapter2_19_0.png" src="../_images/chapter2_19_0.png" />
</div>
</div>
</div>
<div class="section" id="surface-accessibility">
<h3>Surface accessibility<a class="headerlink" href="#surface-accessibility" title="Permalink to this headline">¶</a></h3>
<p>Data describing the solvent-accessible surface of a molecule is of great utility in the
development of that molecule as a therapeutic, particularly in the case of antibodies. In the
context of this report, the most obvious application of molecular surface data is in combination
with the potential sites of chemical modification, described in the previous section. Antibodies
(and other proteins) are known to undergo many different chemical modifications as a result of
interactions with their aqueous environment. The probability and kinetic rate of such a
modification is greatly enhanced by the degree of exposure of the potential modification site to
the solvent environment.
If a 3D structure was supplied or requested by the client, then the graphs below will show the
solvent-accessible surface areas in (Å2) calculated for each residue in the light and heavy chain
variable regions of the antibody. These solvent-accessible surfaces are calculated using the
“rolling ball” algorithm developed by Shrake &amp; Rupley in which a spherical solvent “probe” is3
used, the accessible surface being generated by tracing the center of the probe sphere as it
rolls along the van der Waals surface (as shown below).</p>
<p><img alt="" src="https://upload.wikimedia.org/wikipedia/commons/thumb/d/d3/Accessible_surface.svg/280px-Accessible_surface.svg.png" /></p>
<p align="center">
    Image by Keith Callenberg - courtesy of Wikimedia Common
</p>
<p>The solvent-accessible surface for each residue depends upon the degree of exposure of the
residue on the antibody surface, but also on the size of the residue side chain. Residues that
are completely buried within the hydrophobic core of the antibody, will have solvent-accessible
surface areas at or close to zero. Residues that are highly exposed at the antibody surface will
have values that range from around 100 - 200 Å2, depending upon the size of the residue side
chain and the degree of exposure.</p>
<div class="cell tag_remove-input docutils container">
<div class="cell_output docutils container">
<img alt="../_images/chapter2_21_0.png" src="../_images/chapter2_21_0.png" />
</div>
</div>
</div>
<div class="section" id="instability-index">
<h3>Instability index<a class="headerlink" href="#instability-index" title="Permalink to this headline">¶</a></h3>
<p>The instability index provides an estimate of the stability of your protein in a test tube. Statistical analysis of 12 unstable and 32 stable proteins has revealed that there are certain dipeptides, the occurence of which is significantly different in the unstable proteins compared with those in the stable ones.</p>
<div class="cell tag_remove-input docutils container">
<div class="cell_output docutils container">
<img alt="../_images/chapter2_23_0.png" src="../_images/chapter2_23_0.png" />
</div>
</div>
<hr class="footnotes docutils" />
<dl class="footnote brackets">
<dt class="label" id="footnote1"><span class="brackets"><a class="fn-backref" href="#id1">1</a></span></dt>
<dd><p>The definition for referencing footnotes is generally placed at the bottom of the document.</p>
</dd>
</dl>
</div>
</div>
</div>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "python"
        },
        kernelOptions: {
            kernelName: "python3",
            path: "./ipynb"
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'python3'</script>

              </div>
              
            
                <!-- Previous / next buttons -->
<div class='prev-next-area'> 
    <a class='left-prev' id="prev-link" href="chapter1.html" title="previous page">
        <i class="fas fa-angle-left"></i>
        <div class="prev-next-info">
            <p class="prev-next-subtitle">previous</p>
            <p class="prev-next-title">Protein parameters analysis</p>
        </div>
    </a>
    <a class='right-next' id="next-link" href="chapter3.html" title="next page">
    <div class="prev-next-info">
        <p class="prev-next-subtitle">next</p>
        <p class="prev-next-title">Detection of disulfide bonds</p>
    </div>
    <i class="fas fa-angle-right"></i>
    </a>
</div>
            
        </div>
    </div>
    <footer class="footer">
  <p>
    
      By Nolos bioscience<br/>
    
        &copy; Copyright 2022 Taeyoon Kim.<br/>
  </p>
</footer>
</main>


      </div>
    </div>
  
  <script src="../_static/js/index.be7d3bbb2ef33a8344ce.js"></script>

  </body>
</html>