
<!DOCTYPE html>

<html>
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>Calculation of codon usage &#8212; Afilbercept</title>
    
  <link href="../_static/css/theme.css" rel="stylesheet">
  <link href="../_static/css/index.ff1ffe594081f20da1ef19478df9384b.css" rel="stylesheet">

    
  <link rel="stylesheet"
    href="../_static/vendor/fontawesome/5.13.0/css/all.min.css">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../_static/vendor/fontawesome/5.13.0/webfonts/fa-solid-900.woff2">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../_static/vendor/fontawesome/5.13.0/webfonts/fa-brands-400.woff2">

    
      

    
    <link rel="stylesheet" type="text/css" href="../_static/pygments.css" />
    <link rel="stylesheet" type="text/css" href="../_static/sphinx-book-theme.css?digest=c3fdc42140077d1ad13ad2f1588a4309" />
    <link rel="stylesheet" type="text/css" href="../_static/togglebutton.css" />
    <link rel="stylesheet" type="text/css" href="../_static/copybutton.css" />
    <link rel="stylesheet" type="text/css" href="../_static/mystnb.css" />
    <link rel="stylesheet" type="text/css" href="../_static/sphinx-thebe.css" />
    <link rel="stylesheet" type="text/css" href="../_static/panels-main.c949a650a448cc0ae9fd3441c0e17fb0.css" />
    <link rel="stylesheet" type="text/css" href="../_static/panels-variables.06eb56fa6e07937060861dad626602ad.css" />
    
  <link rel="preload" as="script" href="../_static/js/index.be7d3bbb2ef33a8344ce.js">

    <script data-url_root="../" id="documentation_options" src="../_static/documentation_options.js"></script>
    <script src="../_static/jquery.js"></script>
    <script src="../_static/underscore.js"></script>
    <script src="../_static/doctools.js"></script>
    <script src="../_static/clipboard.min.js"></script>
    <script src="../_static/copybutton.js"></script>
    <script>let toggleHintShow = 'Click to show';</script>
    <script>let toggleHintHide = 'Click to hide';</script>
    <script>let toggleOpenOnPrint = 'true';</script>
    <script src="../_static/togglebutton.js"></script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown, .tag_hide_input div.cell_input, .tag_hide-input div.cell_input, .tag_hide_output div.cell_output, .tag_hide-output div.cell_output, .tag_hide_cell.cell, .tag_hide-cell.cell';</script>
    <script src="../_static/sphinx-book-theme.d59cb220de22ca1c485ebbdc042f0030.js"></script>
    <script>const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"
const thebe_selector = ".thebe,.cell"
const thebe_selector_input = "pre"
const thebe_selector_output = ".output, .cell_output"
</script>
    <script async="async" src="../_static/sphinx-thebe.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <meta name="docsearch:language" content="None">
    

    <!-- Google Analytics -->
    
  </head>
  <body data-spy="scroll" data-target="#bd-toc-nav" data-offset="80">
    
    <div class="container-fluid" id="banner"></div>

    

    <div class="container-xl">
      <div class="row">
          
<div class="col-12 col-md-3 bd-sidebar site-navigation show" id="site-navigation">
    
        <div class="navbar-brand-box">
    <a class="navbar-brand text-wrap" href="../index.html">
      
        <!-- `logo` is deprecated in Sphinx 4.0, so remove this when we stop supporting 3 -->
        
      
      
      <img src="../_static/logo.png" class="logo" alt="logo">
      
      
      <h1 class="site-logo" id="site-title">Afilbercept</h1>
      
    </a>
</div><form class="bd-search d-flex align-items-center" action="../search.html" method="get">
  <i class="icon fas fa-search"></i>
  <input type="search" class="form-control" name="q" id="search-input" placeholder="Search this book..." aria-label="Search this book..." autocomplete="off" >
</form><nav class="bd-links" id="bd-docs-nav" aria-label="Main">
    <div class="bd-toc-item active">
        <ul class="nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="../intro.html">
   About this report
  </a>
 </li>
</ul>
<ul class="nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="chapter1.html">
   Protein parameters analysis
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="chapter2.html">
   Potential sites of chemical modification
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="chapter3.html">
   Detection of disulfide bonds
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="chapter4.html">
   Immunogenicity analysis
  </a>
 </li>
</ul>

    </div>
</nav> <!-- To handle the deprecated key -->

<div class="navbar_extra_footer">
  Powered by <a href="https://jupyterbook.org">Jupyter Book</a>
</div>

</div>


          


          
<main class="col py-md-3 pl-md-4 bd-content overflow-auto" role="main">
    
    <div class="topbar container-xl fixed-top">
    <div class="topbar-contents row">
        <div class="col-12 col-md-3 bd-topbar-whitespace site-navigation show"></div>
        <div class="col pl-md-4 topbar-main">
            
            <button id="navbar-toggler" class="navbar-toggler ml-0" type="button" data-toggle="collapse"
                data-toggle="tooltip" data-placement="bottom" data-target=".site-navigation" aria-controls="navbar-menu"
                aria-expanded="true" aria-label="Toggle navigation" aria-controls="site-navigation"
                title="Toggle navigation" data-toggle="tooltip" data-placement="left">
                <i class="fas fa-bars"></i>
                <i class="fas fa-arrow-left"></i>
                <i class="fas fa-arrow-up"></i>
            </button>
            
            
<div class="dropdown-buttons-trigger">
    <button id="dropdown-buttons-trigger" class="btn btn-secondary topbarbtn" aria-label="Download this page"><i
            class="fas fa-download"></i></button>

    <div class="dropdown-buttons">
        <!-- ipynb file if we had a myst markdown file -->
        
        <!-- Download raw file -->
        <a class="dropdown-buttons" href="../_sources/ipynb/chapter5.ipynb"><button type="button"
                class="btn btn-secondary topbarbtn" title="Download source file" data-toggle="tooltip"
                data-placement="left">.ipynb</button></a>
        <!-- Download PDF via print -->
        <button type="button" id="download-print" class="btn btn-secondary topbarbtn" title="Print to PDF"
                onclick="printPdf(this)" data-toggle="tooltip" data-placement="left">.pdf</button>
    </div>
</div>

            <!-- Source interaction buttons -->

<div class="dropdown-buttons-trigger">
    <button id="dropdown-buttons-trigger" class="btn btn-secondary topbarbtn"
        aria-label="Connect with source repository"><i class="fab fa-github"></i></button>
    <div class="dropdown-buttons sourcebuttons">
        <a class="repository-button"
            href="https://github.com/partrita/ProDevAnalysis"><button type="button" class="btn btn-secondary topbarbtn"
                data-toggle="tooltip" data-placement="left" title="Source repository"><i
                    class="fab fa-github"></i>repository</button></a>
        <a class="issues-button"
            href="https://github.com/partrita/ProDevAnalysis/issues/new?title=Issue%20on%20page%20%2Fipynb/chapter5.html&body=Your%20issue%20content%20here."><button
                type="button" class="btn btn-secondary topbarbtn" data-toggle="tooltip" data-placement="left"
                title="Open an issue"><i class="fas fa-lightbulb"></i>open issue</button></a>
        
    </div>
</div>

            <!-- Full screen (wrap in <a> to have style consistency -->

<a class="full-screen-button"><button type="button" class="btn btn-secondary topbarbtn" data-toggle="tooltip"
        data-placement="bottom" onclick="toggleFullScreen()" aria-label="Fullscreen mode"
        title="Fullscreen mode"><i
            class="fas fa-expand"></i></button></a>

            <!-- Launch buttons -->

<div class="dropdown-buttons-trigger">
    <button id="dropdown-buttons-trigger" class="btn btn-secondary topbarbtn"
        aria-label="Launch interactive content"><i class="fas fa-rocket"></i></button>
    <div class="dropdown-buttons">
        
        <a class="binder-button" href="https://mybinder.org/v2/gh/partrita/ProDevAnalysis/master?urlpath=tree/demo-report/ipynb/chapter5.ipynb"><button type="button"
                class="btn btn-secondary topbarbtn" title="Launch Binder" data-toggle="tooltip"
                data-placement="left"><img class="binder-button-logo"
                    src="../_static/images/logo_binder.svg"
                    alt="Interact on binder">Binder</button></a>
        
        
        
        
    </div>
</div>

        </div>

        <!-- Table of contents -->
        <div class="d-none d-md-block col-md-2 bd-toc show noprint">
            
        </div>
    </div>
</div>
    <div id="main-content" class="row">
        <div class="col-12 col-md-9 pl-md-3 pr-md-0">
            <!-- Table of contents that is only displayed when printing the page -->
            <div id="jb-print-docs-body" class="onlyprint">
                <h1>Calculation of codon usage</h1>
                <!-- Table of contents -->
                <div id="print-main-content">
                    <div id="jb-print-toc">
                        
                    </div>
                </div>
            </div>
            
              <div>
                
  <div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="kn">import</span> <span class="nn">biotite.sequence</span> <span class="k">as</span> <span class="nn">seq</span>
<span class="kn">import</span> <span class="nn">biotite.sequence.align</span> <span class="k">as</span> <span class="nn">align</span>
<span class="kn">import</span> <span class="nn">numpy</span> <span class="k">as</span> <span class="nn">np</span>

<span class="n">alph</span> <span class="o">=</span> <span class="n">seq</span><span class="o">.</span><span class="n">ProteinSequence</span><span class="o">.</span><span class="n">alphabet</span>
<span class="c1"># Load the standard protein substitution matrix, which is BLOSUM62</span>
<span class="n">matrix</span> <span class="o">=</span> <span class="n">align</span><span class="o">.</span><span class="n">SubstitutionMatrix</span><span class="o">.</span><span class="n">std_protein_matrix</span><span class="p">()</span>
<span class="nb">print</span><span class="p">(</span><span class="s2">&quot;</span><span class="se">\n</span><span class="s2">BLOSUM62</span><span class="se">\n</span><span class="s2">&quot;</span><span class="p">)</span>
<span class="nb">print</span><span class="p">(</span><span class="n">matrix</span><span class="p">)</span>
<span class="c1"># Load another matrix from internal database</span>
<span class="n">matrix</span> <span class="o">=</span> <span class="n">align</span><span class="o">.</span><span class="n">SubstitutionMatrix</span><span class="p">(</span><span class="n">alph</span><span class="p">,</span> <span class="n">alph</span><span class="p">,</span> <span class="s2">&quot;BLOSUM50&quot;</span><span class="p">)</span>
<span class="c1"># Load a matrix dictionary representation,</span>
<span class="c1"># modify it, and create the SubstitutionMatrix</span>
<span class="c1"># (The dictionary could be alternatively loaded from a string containing</span>
<span class="c1"># the matrix in NCBI format)</span>
<span class="n">matrix_dict</span> <span class="o">=</span> <span class="n">align</span><span class="o">.</span><span class="n">SubstitutionMatrix</span><span class="o">.</span><span class="n">dict_from_db</span><span class="p">(</span><span class="s2">&quot;BLOSUM62&quot;</span><span class="p">)</span>
<span class="n">matrix_dict</span><span class="p">[(</span><span class="s2">&quot;P&quot;</span><span class="p">,</span><span class="s2">&quot;Y&quot;</span><span class="p">)]</span> <span class="o">=</span> <span class="mi">100</span>
<span class="n">matrix</span> <span class="o">=</span> <span class="n">align</span><span class="o">.</span><span class="n">SubstitutionMatrix</span><span class="p">(</span><span class="n">alph</span><span class="p">,</span> <span class="n">alph</span><span class="p">,</span> <span class="n">matrix_dict</span><span class="p">)</span>
<span class="c1"># And now create a matrix by directly provding the ndarray</span>
<span class="c1"># containing the similarity scores</span>
<span class="c1"># (identity matrix in our case)</span>
<span class="n">scores</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">identity</span><span class="p">(</span><span class="nb">len</span><span class="p">(</span><span class="n">alph</span><span class="p">),</span> <span class="n">dtype</span><span class="o">=</span><span class="nb">int</span><span class="p">)</span>
<span class="n">matrix</span> <span class="o">=</span> <span class="n">align</span><span class="o">.</span><span class="n">SubstitutionMatrix</span><span class="p">(</span><span class="n">alph</span><span class="p">,</span> <span class="n">alph</span><span class="p">,</span> <span class="n">scores</span><span class="p">)</span>
<span class="nb">print</span><span class="p">(</span><span class="s2">&quot;</span><span class="se">\n\n</span><span class="s2">Identity matrix</span><span class="se">\n</span><span class="s2">&quot;</span><span class="p">)</span>
<span class="nb">print</span><span class="p">(</span><span class="n">matrix</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>BLOSUM62

    A   C   D   E   F   G   H   I   K   L   M   N   P   Q   R   S   T   V   W   Y   B   Z   X   *
A   4   0  -2  -1  -2   0  -2  -1  -1  -1  -1  -2  -1  -1  -1   1   0   0  -3  -2  -2  -1   0  -4
C   0   9  -3  -4  -2  -3  -3  -1  -3  -1  -1  -3  -3  -3  -3  -1  -1  -1  -2  -2  -3  -3  -2  -4
D  -2  -3   6   2  -3  -1  -1  -3  -1  -4  -3   1  -1   0  -2   0  -1  -3  -4  -3   4   1  -1  -4
E  -1  -4   2   5  -3  -2   0  -3   1  -3  -2   0  -1   2   0   0  -1  -2  -3  -2   1   4  -1  -4
F  -2  -2  -3  -3   6  -3  -1   0  -3   0   0  -3  -4  -3  -3  -2  -2  -1   1   3  -3  -3  -1  -4
G   0  -3  -1  -2  -3   6  -2  -4  -2  -4  -3   0  -2  -2  -2   0  -2  -3  -2  -3  -1  -2  -1  -4
H  -2  -3  -1   0  -1  -2   8  -3  -1  -3  -2   1  -2   0   0  -1  -2  -3  -2   2   0   0  -1  -4
I  -1  -1  -3  -3   0  -4  -3   4  -3   2   1  -3  -3  -3  -3  -2  -1   3  -3  -1  -3  -3  -1  -4
K  -1  -3  -1   1  -3  -2  -1  -3   5  -2  -1   0  -1   1   2   0  -1  -2  -3  -2   0   1  -1  -4
L  -1  -1  -4  -3   0  -4  -3   2  -2   4   2  -3  -3  -2  -2  -2  -1   1  -2  -1  -4  -3  -1  -4
M  -1  -1  -3  -2   0  -3  -2   1  -1   2   5  -2  -2   0  -1  -1  -1   1  -1  -1  -3  -1  -1  -4
N  -2  -3   1   0  -3   0   1  -3   0  -3  -2   6  -2   0   0   1   0  -3  -4  -2   3   0  -1  -4
P  -1  -3  -1  -1  -4  -2  -2  -3  -1  -3  -2  -2   7  -1  -2  -1  -1  -2  -4  -3  -2  -1  -2  -4
Q  -1  -3   0   2  -3  -2   0  -3   1  -2   0   0  -1   5   1   0  -1  -2  -2  -1   0   3  -1  -4
R  -1  -3  -2   0  -3  -2   0  -3   2  -2  -1   0  -2   1   5  -1  -1  -3  -3  -2  -1   0  -1  -4
S   1  -1   0   0  -2   0  -1  -2   0  -2  -1   1  -1   0  -1   4   1  -2  -3  -2   0   0   0  -4
T   0  -1  -1  -1  -2  -2  -2  -1  -1  -1  -1   0  -1  -1  -1   1   5   0  -2  -2  -1  -1   0  -4
V   0  -1  -3  -2  -1  -3  -3   3  -2   1   1  -3  -2  -2  -3  -2   0   4  -3  -1  -3  -2  -1  -4
W  -3  -2  -4  -3   1  -2  -2  -3  -3  -2  -1  -4  -4  -2  -3  -3  -2  -3  11   2  -4  -3  -2  -4
Y  -2  -2  -3  -2   3  -3   2  -1  -2  -1  -1  -2  -3  -1  -2  -2  -2  -1   2   7  -3  -2  -1  -4
B  -2  -3   4   1  -3  -1   0  -3   0  -4  -3   3  -2   0  -1   0  -1  -3  -4  -3   4   1  -1  -4
Z  -1  -3   1   4  -3  -2   0  -3   1  -3  -1   0  -1   3   0   0  -1  -2  -3  -2   1   4  -1  -4
X   0  -2  -1  -1  -1  -1  -1  -1  -1  -1  -1  -1  -2  -1  -1   0   0  -1  -2  -1  -1  -1  -1  -4
*  -4  -4  -4  -4  -4  -4  -4  -4  -4  -4  -4  -4  -4  -4  -4  -4  -4  -4  -4  -4  -4  -4  -4   1


Identity matrix

    A   C   D   E   F   G   H   I   K   L   M   N   P   Q   R   S   T   V   W   Y   B   Z   X   *
A   1   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0
C   0   1   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0
D   0   0   1   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0
E   0   0   0   1   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0
F   0   0   0   0   1   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0
G   0   0   0   0   0   1   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0
H   0   0   0   0   0   0   1   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0
I   0   0   0   0   0   0   0   1   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0
K   0   0   0   0   0   0   0   0   1   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0
L   0   0   0   0   0   0   0   0   0   1   0   0   0   0   0   0   0   0   0   0   0   0   0   0
M   0   0   0   0   0   0   0   0   0   0   1   0   0   0   0   0   0   0   0   0   0   0   0   0
N   0   0   0   0   0   0   0   0   0   0   0   1   0   0   0   0   0   0   0   0   0   0   0   0
P   0   0   0   0   0   0   0   0   0   0   0   0   1   0   0   0   0   0   0   0   0   0   0   0
Q   0   0   0   0   0   0   0   0   0   0   0   0   0   1   0   0   0   0   0   0   0   0   0   0
R   0   0   0   0   0   0   0   0   0   0   0   0   0   0   1   0   0   0   0   0   0   0   0   0
S   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   1   0   0   0   0   0   0   0   0
T   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   1   0   0   0   0   0   0   0
V   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   1   0   0   0   0   0   0
W   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   1   0   0   0   0   0
Y   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   1   0   0   0   0
B   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   1   0   0   0
Z   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   1   0   0
X   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   1   0
*   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   1
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">seq1</span> <span class="o">=</span> <span class="n">seq</span><span class="o">.</span><span class="n">ProteinSequence</span><span class="p">(</span><span class="s2">&quot;BIQTITE&quot;</span><span class="p">)</span>
<span class="n">seq2</span> <span class="o">=</span> <span class="n">seq</span><span class="o">.</span><span class="n">ProteinSequence</span><span class="p">(</span><span class="s2">&quot;IQLITE&quot;</span><span class="p">)</span>
<span class="n">matrix</span> <span class="o">=</span> <span class="n">align</span><span class="o">.</span><span class="n">SubstitutionMatrix</span><span class="o">.</span><span class="n">std_protein_matrix</span><span class="p">()</span>
<span class="nb">print</span><span class="p">(</span><span class="s2">&quot;</span><span class="se">\n</span><span class="s2">Local alignment&quot;</span><span class="p">)</span>
<span class="n">alignments</span> <span class="o">=</span> <span class="n">align</span><span class="o">.</span><span class="n">align_optimal</span><span class="p">(</span><span class="n">seq1</span><span class="p">,</span> <span class="n">seq2</span><span class="p">,</span> <span class="n">matrix</span><span class="p">,</span> <span class="n">local</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<span class="k">for</span> <span class="n">ali</span> <span class="ow">in</span> <span class="n">alignments</span><span class="p">:</span>
    <span class="nb">print</span><span class="p">(</span><span class="n">ali</span><span class="p">)</span>
<span class="nb">print</span><span class="p">(</span><span class="s2">&quot;Global alignment&quot;</span><span class="p">)</span>
<span class="n">alignments</span> <span class="o">=</span> <span class="n">align</span><span class="o">.</span><span class="n">align_optimal</span><span class="p">(</span><span class="n">seq1</span><span class="p">,</span> <span class="n">seq2</span><span class="p">,</span> <span class="n">matrix</span><span class="p">,</span> <span class="n">local</span><span class="o">=</span><span class="kc">False</span><span class="p">)</span>
<span class="k">for</span> <span class="n">ali</span> <span class="ow">in</span> <span class="n">alignments</span><span class="p">:</span>
    <span class="nb">print</span><span class="p">(</span><span class="n">ali</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Local alignment
IQTITE
IQLITE
Global alignment
BIQTITE
-IQLITE
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="kn">import</span> <span class="nn">matplotlib.pyplot</span> <span class="k">as</span> <span class="nn">plt</span>
<span class="kn">import</span> <span class="nn">biotite.sequence.graphics</span> <span class="k">as</span> <span class="nn">graphics</span>

<span class="n">fig</span><span class="p">,</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">(</span><span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mf">2.0</span><span class="p">,</span> <span class="mf">0.8</span><span class="p">))</span>
<span class="n">graphics</span><span class="o">.</span><span class="n">plot_alignment_similarity_based</span><span class="p">(</span>
    <span class="n">ax</span><span class="p">,</span> <span class="n">alignments</span><span class="p">[</span><span class="mi">0</span><span class="p">],</span> <span class="n">matrix</span><span class="o">=</span><span class="n">matrix</span><span class="p">,</span> <span class="n">symbols_per_line</span><span class="o">=</span><span class="nb">len</span><span class="p">(</span><span class="n">alignments</span><span class="p">[</span><span class="mi">0</span><span class="p">])</span>
<span class="p">)</span>
<span class="n">fig</span><span class="o">.</span><span class="n">tight_layout</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="../_images/chapter5_2_0.png" src="../_images/chapter5_2_0.png" />
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="kn">import</span> <span class="nn">matplotlib.pyplot</span> <span class="k">as</span> <span class="nn">plt</span>
<span class="kn">import</span> <span class="nn">biotite.sequence</span> <span class="k">as</span> <span class="nn">seq</span>
<span class="kn">import</span> <span class="nn">biotite.sequence.align</span> <span class="k">as</span> <span class="nn">align</span>
<span class="kn">import</span> <span class="nn">biotite.sequence.io.fasta</span> <span class="k">as</span> <span class="nn">fasta</span>
<span class="kn">import</span> <span class="nn">biotite.database.entrez</span> <span class="k">as</span> <span class="nn">entrez</span>
<span class="kn">import</span> <span class="nn">biotite.sequence.graphics</span> <span class="k">as</span> <span class="nn">graphics</span>

<span class="c1"># Download and parse protein sequences of avidin and streptavidin</span>
<span class="n">fasta_file</span> <span class="o">=</span> <span class="n">fasta</span><span class="o">.</span><span class="n">FastaFile</span><span class="o">.</span><span class="n">read</span><span class="p">(</span><span class="n">entrez</span><span class="o">.</span><span class="n">fetch_single_file</span><span class="p">(</span>
    <span class="p">[</span><span class="s2">&quot;CAC34569&quot;</span><span class="p">,</span> <span class="s2">&quot;ACL82594&quot;</span><span class="p">],</span> <span class="kc">None</span><span class="p">,</span> <span class="s2">&quot;protein&quot;</span><span class="p">,</span> <span class="s2">&quot;fasta&quot;</span>
<span class="p">))</span>
<span class="k">for</span> <span class="n">name</span><span class="p">,</span> <span class="n">sequence</span> <span class="ow">in</span> <span class="n">fasta_file</span><span class="o">.</span><span class="n">items</span><span class="p">():</span>
    <span class="k">if</span> <span class="s2">&quot;CAC34569&quot;</span> <span class="ow">in</span> <span class="n">name</span><span class="p">:</span>
        <span class="n">avidin_seq</span> <span class="o">=</span> <span class="n">seq</span><span class="o">.</span><span class="n">ProteinSequence</span><span class="p">(</span><span class="n">sequence</span><span class="p">)</span>
    <span class="k">elif</span> <span class="s2">&quot;ACL82594&quot;</span> <span class="ow">in</span> <span class="n">name</span><span class="p">:</span>
        <span class="n">streptavidin_seq</span> <span class="o">=</span> <span class="n">seq</span><span class="o">.</span><span class="n">ProteinSequence</span><span class="p">(</span><span class="n">sequence</span><span class="p">)</span>

<span class="c1"># Get BLOSUM62 matrix</span>
<span class="n">matrix</span> <span class="o">=</span> <span class="n">align</span><span class="o">.</span><span class="n">SubstitutionMatrix</span><span class="o">.</span><span class="n">std_protein_matrix</span><span class="p">()</span>
<span class="c1"># Perform pairwise sequence alignment with affine gap penalty</span>
<span class="c1"># Terminal gaps are not penalized</span>
<span class="n">alignments</span> <span class="o">=</span> <span class="n">align</span><span class="o">.</span><span class="n">align_optimal</span><span class="p">(</span><span class="n">avidin_seq</span><span class="p">,</span> <span class="n">streptavidin_seq</span><span class="p">,</span> <span class="n">matrix</span><span class="p">,</span>
                                 <span class="n">gap_penalty</span><span class="o">=</span><span class="p">(</span><span class="o">-</span><span class="mi">10</span><span class="p">,</span> <span class="o">-</span><span class="mi">1</span><span class="p">),</span> <span class="n">terminal_penalty</span><span class="o">=</span><span class="kc">False</span><span class="p">)</span>

<span class="c1"># Draw first and only alignment</span>
<span class="c1"># The color intensity indicates the similiarity</span>
<span class="n">fig</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">figure</span><span class="p">(</span><span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mf">8.0</span><span class="p">,</span> <span class="mf">2.5</span><span class="p">))</span>
<span class="n">ax</span> <span class="o">=</span> <span class="n">fig</span><span class="o">.</span><span class="n">add_subplot</span><span class="p">(</span><span class="mi">111</span><span class="p">)</span>
<span class="n">graphics</span><span class="o">.</span><span class="n">plot_alignment_similarity_based</span><span class="p">(</span>
    <span class="n">ax</span><span class="p">,</span> <span class="n">alignments</span><span class="p">[</span><span class="mi">0</span><span class="p">],</span> <span class="n">matrix</span><span class="o">=</span><span class="n">matrix</span><span class="p">,</span> <span class="n">labels</span><span class="o">=</span><span class="p">[</span><span class="s2">&quot;Avidin&quot;</span><span class="p">,</span> <span class="s2">&quot;Streptavidin&quot;</span><span class="p">],</span>
    <span class="n">show_numbers</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span> <span class="n">show_line_position</span><span class="o">=</span><span class="kc">True</span>
<span class="p">)</span>
<span class="n">fig</span><span class="o">.</span><span class="n">tight_layout</span><span class="p">()</span>

<span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="../_images/chapter5_3_0.png" src="../_images/chapter5_3_0.png" />
</div>
</div>
<div class="tex2jax_ignore mathjax_ignore section" id="calculation-of-codon-usage">
<h1>Calculation of codon usage<a class="headerlink" href="#calculation-of-codon-usage" title="Permalink to this headline">¶</a></h1>
<p>This script creates a table for codon usage in the Escherichia coli K-12 strain. Codon usage describes the frequencies of the codons that code for an amino acid. These frequencies are expected to reflect the abundance of the respective tRNAs in the target organism. The codon usage differs from species to species. Hence, it is important to look at the codon usage of the organism of interest for applications like codon optimization etc.</p>
<p>For the computation of the codon usage we will have a look into the annotated Escherichia coli K-12 genome: The script extracts all coding sequences (CDS) from the genome and counts the total number of each codon in these sequences. Then the relative frequencies are calculated by dividing the total number of occurrences of each codon by the total number of occurrences of the respective amino acid. In order to improve the performance, the script mostly works with symbol codes (see tutorial) instead of the symbols itself.</p>
<p>At first we fetch the genome from the NCBI Entrez database (Accession: U00096) as GenBank file and parse it into an AnnotatedSequence. Then we create a dictionary that will store the total codon frequencies later on. As already mentioned, the script works with symbol codes. Consequently, each codon in the dictionary is described as 3 integers instead of 3 letters.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Code source: Patrick Kunzmann</span>
<span class="c1"># License: BSD 3 clause</span>

<span class="kn">import</span> <span class="nn">tempfile</span>
<span class="kn">import</span> <span class="nn">itertools</span>
<span class="kn">import</span> <span class="nn">numpy</span> <span class="k">as</span> <span class="nn">np</span>
<span class="kn">import</span> <span class="nn">biotite.sequence</span> <span class="k">as</span> <span class="nn">seq</span>
<span class="kn">import</span> <span class="nn">biotite.sequence.io.genbank</span> <span class="k">as</span> <span class="nn">gb</span>
<span class="kn">import</span> <span class="nn">biotite.sequence.io.fasta</span> <span class="k">as</span> <span class="nn">fasta</span>
<span class="kn">import</span> <span class="nn">biotite.database.entrez</span> <span class="k">as</span> <span class="nn">entrez</span>


<span class="c1"># Get the E. coli K-12 genome as annotated sequence</span>
<span class="n">gb_file</span> <span class="o">=</span> <span class="n">gb</span><span class="o">.</span><span class="n">GenBankFile</span><span class="o">.</span><span class="n">read</span><span class="p">(</span>
    <span class="n">entrez</span><span class="o">.</span><span class="n">fetch</span><span class="p">(</span><span class="s2">&quot;U00096&quot;</span><span class="p">,</span> <span class="n">tempfile</span><span class="o">.</span><span class="n">gettempdir</span><span class="p">(),</span> <span class="s2">&quot;gb&quot;</span><span class="p">,</span> <span class="s2">&quot;nuccore&quot;</span><span class="p">,</span> <span class="s2">&quot;gb&quot;</span><span class="p">)</span>
<span class="p">)</span>
<span class="c1"># We are only interested in CDS features</span>
<span class="n">k12_genome</span> <span class="o">=</span> <span class="n">gb</span><span class="o">.</span><span class="n">get_annotated_sequence</span><span class="p">(</span><span class="n">gb_file</span><span class="p">,</span> <span class="n">include_only</span><span class="o">=</span><span class="p">[</span><span class="s2">&quot;CDS&quot;</span><span class="p">])</span>


<span class="c1"># This dictionary will count how often each codon occurs in the genome</span>
<span class="c1"># For increased performance the dictionary uses symbol codes ([0 3 2])</span>
<span class="c1"># instead of symbols ([&#39;A&#39; &#39;T&#39; &#39;G&#39;]) as keys</span>
<span class="n">codon_counter</span> <span class="o">=</span> <span class="p">{</span>
    <span class="n">codon</span><span class="p">:</span> <span class="mi">0</span> <span class="k">for</span> <span class="n">codon</span>
    <span class="ow">in</span> <span class="n">itertools</span><span class="o">.</span><span class="n">product</span><span class="p">(</span> <span class="o">*</span><span class="p">([</span><span class="nb">range</span><span class="p">(</span><span class="nb">len</span><span class="p">(</span><span class="n">k12_genome</span><span class="o">.</span><span class="n">sequence</span><span class="o">.</span><span class="n">alphabet</span><span class="p">))]</span> <span class="o">*</span> <span class="mi">3</span><span class="p">)</span> <span class="p">)</span>
<span class="p">}</span>
<span class="c1"># For demonstration purposes print the 64 codons in symbol code form</span>
<span class="c1"># print(list(codon_counter.keys()))</span>

<span class="c1"># Iterate over all CDS features</span>
<span class="k">for</span> <span class="n">cds</span> <span class="ow">in</span> <span class="n">k12_genome</span><span class="o">.</span><span class="n">annotation</span><span class="p">:</span>
    <span class="c1"># Obtain the sequence of the CDS</span>
    <span class="n">cds_seq</span> <span class="o">=</span> <span class="n">k12_genome</span><span class="p">[</span><span class="n">cds</span><span class="p">]</span>
    <span class="k">if</span> <span class="nb">len</span><span class="p">(</span><span class="n">cds_seq</span><span class="p">)</span> <span class="o">%</span> <span class="mi">3</span> <span class="o">!=</span> <span class="mi">0</span><span class="p">:</span>
        <span class="c1"># A CDS&#39; length should be a multiple of 3,</span>
        <span class="c1"># otherwise the CDS is malformed</span>
        <span class="k">continue</span>
    <span class="c1"># Iterate over the sequence in non-overlapping frames of 3</span>
    <span class="c1"># and count the occurence of each codon</span>
    <span class="k">for</span> <span class="n">i</span> <span class="ow">in</span> <span class="nb">range</span><span class="p">(</span><span class="mi">0</span><span class="p">,</span> <span class="nb">len</span><span class="p">(</span><span class="n">cds_seq</span><span class="p">),</span> <span class="mi">3</span><span class="p">):</span>
        <span class="n">codon_code</span> <span class="o">=</span> <span class="nb">tuple</span><span class="p">(</span><span class="n">cds_seq</span><span class="o">.</span><span class="n">code</span><span class="p">[</span><span class="n">i</span><span class="p">:</span><span class="n">i</span><span class="o">+</span><span class="mi">3</span><span class="p">])</span>
        <span class="n">codon_counter</span><span class="p">[</span><span class="n">codon_code</span><span class="p">]</span> <span class="o">+=</span> <span class="mi">1</span>

<span class="c1"># Convert the total frequencies into relative frequencies</span>
<span class="c1"># for each amino acid</span>
<span class="c1"># The NCBI codon table with ID 11 is the bacterial codon table</span>
<span class="n">table</span> <span class="o">=</span> <span class="n">seq</span><span class="o">.</span><span class="n">CodonTable</span><span class="o">.</span><span class="n">load</span><span class="p">(</span><span class="mi">11</span><span class="p">)</span>
<span class="c1"># As the script uses symbol codes, each amino acid is represented by a</span>
<span class="c1"># number between 0 and 19, instead of the single letter code</span>
<span class="k">for</span> <span class="n">amino_acid_code</span> <span class="ow">in</span> <span class="nb">range</span><span class="p">(</span><span class="mi">20</span><span class="p">):</span>
    <span class="c1"># Find all codons coding for the amino acid</span>
    <span class="c1"># The codons are also in symbol code format, e.g. ATG -&gt; (0, 3, 2)</span>
    <span class="n">codon_codes_for_aa</span> <span class="o">=</span> <span class="n">table</span><span class="p">[</span><span class="n">amino_acid_code</span><span class="p">]</span>
    <span class="c1"># Get the total amount of codon occurrences for the amino acid</span>
    <span class="n">total</span> <span class="o">=</span> <span class="mi">0</span>
    <span class="k">for</span> <span class="n">codon_code</span> <span class="ow">in</span> <span class="n">codon_codes_for_aa</span><span class="p">:</span>
        <span class="n">total</span> <span class="o">+=</span> <span class="n">codon_counter</span><span class="p">[</span><span class="n">codon_code</span><span class="p">]</span>
    <span class="c1"># Replace the total frequencies with relative frequencies</span>
    <span class="c1"># and print it</span>
    <span class="k">for</span> <span class="n">codon_code</span> <span class="ow">in</span> <span class="n">codon_codes_for_aa</span><span class="p">:</span>
        <span class="c1"># Convert total frequencies into relative frequencies</span>
        <span class="n">codon_counter</span><span class="p">[</span><span class="n">codon_code</span><span class="p">]</span> <span class="o">/=</span> <span class="n">total</span>
        <span class="c1"># The rest of this code block prints the codon usage table</span>
        <span class="c1"># in human readable format</span>
        <span class="n">amino_acid</span> <span class="o">=</span> <span class="n">seq</span><span class="o">.</span><span class="n">ProteinSequence</span><span class="o">.</span><span class="n">alphabet</span><span class="o">.</span><span class="n">decode</span><span class="p">(</span><span class="n">amino_acid_code</span><span class="p">)</span>
        <span class="c1"># Convert the symbol codes for each codon into symbols...</span>
        <span class="c1"># ([0,3,2] -&gt; [&#39;A&#39; &#39;T&#39; &#39;G&#39;])</span>
        <span class="n">codon</span> <span class="o">=</span> <span class="n">k12_genome</span><span class="o">.</span><span class="n">sequence</span><span class="o">.</span><span class="n">alphabet</span><span class="o">.</span><span class="n">decode_multiple</span><span class="p">(</span><span class="n">codon_code</span><span class="p">)</span>
        <span class="c1"># ...and represent as string</span>
        <span class="c1"># ([&#39;A&#39; &#39;T&#39; &#39;G&#39;] -&gt; &quot;ATG&quot;)</span>
        <span class="n">codon</span> <span class="o">=</span> <span class="s2">&quot;&quot;</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">codon</span><span class="p">)</span>
        <span class="n">freq</span> <span class="o">=</span> <span class="n">codon_counter</span><span class="p">[</span><span class="n">codon_code</span><span class="p">]</span>
        <span class="nb">print</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;</span><span class="si">{</span><span class="n">amino_acid</span><span class="si">}</span><span class="s2">   </span><span class="si">{</span><span class="n">codon</span><span class="si">}</span><span class="s2">   </span><span class="si">{</span><span class="n">freq</span><span class="si">:</span><span class="s2">.2f</span><span class="si">}</span><span class="s2">&quot;</span><span class="p">)</span>
    <span class="nb">print</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>A   GCA   0.21
A   GCC   0.27
A   GCG   0.36
A   GCT   0.16

C   TGC   0.56
C   TGT   0.44

D   GAC   0.37
D   GAT   0.63

E   GAA   0.69
E   GAG   0.31

F   TTC   0.43
F   TTT   0.57

G   GGA   0.11
G   GGC   0.41
G   GGG   0.15
G   GGT   0.34

H   CAC   0.43
H   CAT   0.57

I   ATA   0.07
I   ATC   0.42
I   ATT   0.51

K   AAA   0.77
K   AAG   0.23

L   CTA   0.04
L   CTC   0.10
L   CTG   0.50
L   CTT   0.10
L   TTA   0.13
L   TTG   0.13

M   ATG   1.00

N   AAC   0.55
N   AAT   0.45

P   CCA   0.19
P   CCC   0.12
P   CCG   0.53
P   CCT   0.16

Q   CAA   0.35
Q   CAG   0.65

R   AGA   0.04
R   AGG   0.02
R   CGA   0.06
R   CGC   0.40
R   CGG   0.10
R   CGT   0.38

S   AGC   0.28
S   AGT   0.15
S   TCA   0.12
S   TCC   0.15
S   TCG   0.15
S   TCT   0.15

T   ACA   0.13
T   ACC   0.44
T   ACG   0.27
T   ACT   0.17

V   GTA   0.15
V   GTC   0.22
V   GTG   0.37
V   GTT   0.26

W   TGG   1.00

Y   TAC   0.43
Y   TAT   0.57
</pre></div>
</div>
</div>
</div>
<p>As expected the dictionary encodes each codon as tuple of 3 numbers, where 0 represents A, 1 C, 2 G and 3 T. These mappings are defined by the alphabet of the genomic sequence.</p>
<p>In the next step the occurrences of codons in the coding sequences are counted, the relative frequencies are calculated and the codon table is printed.</p>
<p>The codon usage table can be used to optimize recombinant protein expression by designing the DNA sequence for the target protein according to the codon usage. This is called codon optimization. However, there is a variety of different algorithms for codon optimization. For simplicity reasons, this example uses an approach, where for every amino acid always the most frequently occuring codon is used.</p>
<p>In the follwing we will derive a codon optimized DNA sequence of streptavidin for expression in E. coli K-12.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># For each amino acid, select the codon with maximum codon usage</span>
<span class="c1"># Again, symbol codes are used here</span>
<span class="n">opt_codons</span> <span class="o">=</span> <span class="p">{}</span>
<span class="k">for</span> <span class="n">amino_acid_code</span> <span class="ow">in</span> <span class="nb">range</span><span class="p">(</span><span class="mi">20</span><span class="p">):</span>
    <span class="n">codon_codes_for_aa</span> <span class="o">=</span> <span class="n">table</span><span class="p">[</span><span class="n">amino_acid_code</span><span class="p">]</span>
    <span class="c1"># Find codon with maximum frequency</span>
    <span class="n">max_freq</span> <span class="o">=</span> <span class="mi">0</span>
    <span class="n">best_codon_code</span> <span class="o">=</span> <span class="kc">None</span>
    <span class="k">for</span> <span class="n">codon_code</span> <span class="ow">in</span> <span class="n">codon_codes_for_aa</span><span class="p">:</span>
        <span class="k">if</span> <span class="n">codon_counter</span><span class="p">[</span><span class="n">codon_code</span><span class="p">]</span> <span class="o">&gt;</span> <span class="n">max_freq</span><span class="p">:</span>
            <span class="n">max_freq</span> <span class="o">=</span> <span class="n">codon_counter</span><span class="p">[</span><span class="n">codon_code</span><span class="p">]</span>
            <span class="n">best_codon_code</span> <span class="o">=</span> <span class="n">codon_code</span>
    <span class="c1"># Map the amino acid to the codon with maximum frequency</span>
    <span class="n">opt_codons</span><span class="p">[</span><span class="n">amino_acid_code</span><span class="p">]</span> <span class="o">=</span> <span class="n">best_codon_code</span>

<span class="c1"># Fetch the streptavidin protein sequence from Streptomyces avidinii</span>
<span class="n">fasta_file</span> <span class="o">=</span> <span class="n">fasta</span><span class="o">.</span><span class="n">FastaFile</span><span class="o">.</span><span class="n">read</span><span class="p">(</span>
    <span class="n">entrez</span><span class="o">.</span><span class="n">fetch</span><span class="p">(</span><span class="s2">&quot;P22629&quot;</span><span class="p">,</span> <span class="kc">None</span><span class="p">,</span> <span class="s2">&quot;fasta&quot;</span><span class="p">,</span> <span class="s2">&quot;protein&quot;</span><span class="p">,</span> <span class="s2">&quot;fasta&quot;</span><span class="p">)</span>
<span class="p">)</span>
<span class="n">strep_prot_seq</span> <span class="o">=</span> <span class="n">fasta</span><span class="o">.</span><span class="n">get_sequence</span><span class="p">(</span><span class="n">fasta_file</span><span class="p">)</span>
<span class="c1"># Create a DNA sequence from the protein sequence</span>
<span class="c1"># using the optimal codons</span>
<span class="n">strep_dna_seq</span> <span class="o">=</span> <span class="n">seq</span><span class="o">.</span><span class="n">NucleotideSequence</span><span class="p">()</span>
<span class="n">strep_dna_seq</span><span class="o">.</span><span class="n">code</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">concatenate</span><span class="p">(</span>
    <span class="p">[</span><span class="n">opt_codons</span><span class="p">[</span><span class="n">amino_acid_code</span><span class="p">]</span> <span class="k">for</span> <span class="n">amino_acid_code</span> <span class="ow">in</span> <span class="n">strep_prot_seq</span><span class="o">.</span><span class="n">code</span><span class="p">]</span>
<span class="p">)</span>
<span class="c1"># Add stop codon</span>
<span class="n">strep_dna_seq</span> <span class="o">+=</span> <span class="n">seq</span><span class="o">.</span><span class="n">NucleotideSequence</span><span class="p">(</span><span class="s2">&quot;TAA&quot;</span><span class="p">)</span>
<span class="c1"># Put the DNA sequence into a FASTA file</span>
<span class="n">fasta_file</span> <span class="o">=</span> <span class="n">fasta</span><span class="o">.</span><span class="n">FastaFile</span><span class="p">()</span>
<span class="n">fasta_file</span><span class="p">[</span><span class="s2">&quot;Codon optimized sequence&quot;</span><span class="p">]</span> <span class="o">=</span> <span class="nb">str</span><span class="p">(</span><span class="n">strep_dna_seq</span><span class="p">)</span>
<span class="c1"># Print the contents of the created FASTA file</span>
<span class="nb">print</span><span class="p">(</span><span class="n">fasta_file</span><span class="p">)</span>
<span class="c1"># In a real application it would be written onto the hard drive via</span>
<span class="c1"># fasta_file.write(&quot;../data/some_file.fasta&quot;)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>&gt;Codon optimized sequence
ATGCGCAAAATTGTGGTGGCGGCGATTGCGGTGAGCCTGACCACCGTGAGCATTACCGCGAGCGCGAGCGCGGATCCGAG
CAAAGATAGCAAAGCGCAGGTGAGCGCGGCGGAAGCGGGCATTACCGGCACCTGGTATAACCAGCTGGGCAGCACCTTTA
TTGTGACCGCGGGCGCGGATGGCGCGCTGACCGGCACCTATGAAAGCGCGGTGGGCAACGCGGAAAGCCGCTATGTGCTG
ACCGGCCGCTATGATAGCGCGCCGGCGACCGATGGCAGCGGCACCGCGCTGGGCTGGACCGTGGCGTGGAAAAACAACTA
TCGCAACGCGCATAGCGCGACCACCTGGAGCGGCCAGTATGTGGGCGGCGCGGAAGCGCGCATTAACACCCAGTGGCTGC
TGACCAGCGGCACCACCGAAGCGAACGCGTGGAAAAGCACCCTGGTGGGCCATGATACCTTTACCAAAGTGAAACCGAGC
GCGGCGAGCATTGATGCGGCGAAAAAAGCGGGCGTGAACAACGGCAACCCGCTGGATGCGGTGCAGCAGTAA
</pre></div>
</div>
</div>
</div>
</div>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "python"
        },
        kernelOptions: {
            kernelName: "python3",
            path: "./ipynb"
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'python3'</script>

              </div>
              
            
                <!-- Previous / next buttons -->
<div class='prev-next-area'>
</div>
            
        </div>
    </div>
    <footer class="footer">
  <p>
    
      By Nolos bioscience<br/>
    
        &copy; Copyright 2022 Taeyoon Kim.<br/>
  </p>
</footer>
</main>


      </div>
    </div>
  
  <script src="../_static/js/index.be7d3bbb2ef33a8344ce.js"></script>

  </body>
</html>