%% Generated by Sphinx.
\def\sphinxdocclass{jupyterBook}
\documentclass[letterpaper,10pt,english]{jupyterBook}
\ifdefined\pdfpxdimen
   \let\sphinxpxdimen\pdfpxdimen\else\newdimen\sphinxpxdimen
\fi \sphinxpxdimen=.75bp\relax
\ifdefined\pdfimageresolution
    \pdfimageresolution= \numexpr \dimexpr1in\relax/\sphinxpxdimen\relax
\fi
%% let collapsible pdf bookmarks panel have high depth per default
\PassOptionsToPackage{bookmarksdepth=5}{hyperref}
%% turn off hyperref patch of \index as sphinx.xdy xindy module takes care of
%% suitable \hyperpage mark-up, working around hyperref-xindy incompatibility
\PassOptionsToPackage{hyperindex=false}{hyperref}
%% memoir class requires extra handling
\makeatletter\@ifclassloaded{memoir}
{\ifdefined\memhyperindexfalse\memhyperindexfalse\fi}{}\makeatother

\PassOptionsToPackage{warn}{textcomp}

\catcode`^^^^00a0\active\protected\def^^^^00a0{\leavevmode\nobreak\ }
\usepackage{cmap}
\usepackage{fontspec}
\defaultfontfeatures[\rmfamily,\sffamily,\ttfamily]{}
\usepackage{amsmath,amssymb,amstext}
\usepackage{polyglossia}
\setmainlanguage{english}



\setmainfont{FreeSerif}[
  Extension      = .otf,
  UprightFont    = *,
  ItalicFont     = *Italic,
  BoldFont       = *Bold,
  BoldItalicFont = *BoldItalic
]
\setsansfont{FreeSans}[
  Extension      = .otf,
  UprightFont    = *,
  ItalicFont     = *Oblique,
  BoldFont       = *Bold,
  BoldItalicFont = *BoldOblique,
]
\setmonofont{FreeMono}[
  Extension      = .otf,
  UprightFont    = *,
  ItalicFont     = *Oblique,
  BoldFont       = *Bold,
  BoldItalicFont = *BoldOblique,
]



\usepackage[Bjarne]{fncychap}
\usepackage[,numfigreset=1,mathnumfig]{sphinx}

\fvset{fontsize=\small}
\usepackage{geometry}


% Include hyperref last.
\usepackage{hyperref}
% Fix anchor placement for figures with captions.
\usepackage{hypcap}% it must be loaded after hyperref.
% Set up styles of URL: it should be placed after hyperref.
\urlstyle{same}


\usepackage{sphinxmessages}



        % Start of preamble defined in sphinx-jupyterbook-latex %
         \usepackage[Latin,Greek]{ucharclasses}
        \usepackage{unicode-math}
        % fixing title of the toc
        \addto\captionsenglish{\renewcommand{\contentsname}{Contents}}
        \hypersetup{
            pdfencoding=auto,
            psdextra
        }
        % End of preamble defined in sphinx-jupyterbook-latex %
        

\title{Afilbercept}
\date{May 12, 2022}
\release{}
\author{Taeyoon kim}
\newcommand{\sphinxlogo}{\vbox{}}
\renewcommand{\releasename}{}
\makeindex
\begin{document}

\pagestyle{empty}
\sphinxmaketitle
\pagestyle{plain}
\sphinxtableofcontents
\pagestyle{normal}
\phantomsection\label{\detokenize{intro::doc}}


\sphinxAtStartPar
The data in the report is intended to be a resource to guide the development and lead\sphinxhyphen{}optimization of a clinical antibody. These data can be used in a preemptive fashion \sphinxhyphen{} for example, in the decision to substitute an exposed residue on the antibody surface that may be prone to the kind of chemical modification that might affect the stability of the antibody. They can also be used to assist the troubleshooting of problems that can arise in the course of an antibody’s clinical development \sphinxhyphen{} for example if an antibody displays stability issues in storage, or unacceptably high levels of immunogenicity in early clinical trials. There is always a great deal of risk involved in the development of any therapeutic molecule but experience has shown that the kind of data presented in this report is an invaluable tool for mitigating that risk \sphinxhyphen{} either by helping to identify potential problems before they occur, or by guiding the troubleshooting ofproblems that can occur during the antibody’s development and lead\sphinxhyphen{}optimization.

\begin{DUlineblock}{0em}
\item[] \sphinxstylestrong{\Large Introduction}
\end{DUlineblock}

\sphinxAtStartPar
This report contains a structure and sequence\sphinxhyphen{}based analysis of Aflibercept.
\begin{quote}

\sphinxAtStartPar
Aflibercept, sold under the brand names Eylea and Zaltrap, is a medication used to treat wet macular degeneration and metastatic colorectal cancer. It was developed by Regeneron Pharmaceuticals and is approved in the United States and the European Union. –wiki
\end{quote}

\sphinxAtStartPar
Aflibercept is a recombinant fusion protein consisting of vascular endothelial growth factor (VEGF)\sphinxhyphen{}binding portions from the extracellular domains of human VEGF receptors 1 and 2, that are fused to the Fc portion of the human IgG1 immunoglobulin.

\begin{DUlineblock}{0em}
\item[] \sphinxstylestrong{\large History}
\end{DUlineblock}

\sphinxAtStartPar
Regeneron commenced clinical testing of aflibercept in cancer in 2001. In 2003, Regeneron signed a major deal with Aventis to develop aflibercept in the field of cancer. In 2004 Regeneron started testing the compound, locally delivered, in proliferative eye diseases, and in 2006 Regeneron and Bayer signed an agreement to develop the eye indications.

\begin{DUlineblock}{0em}
\item[] \sphinxstylestrong{\large Table of contents}
\end{DUlineblock}
\begin{itemize}
\item {} 
\sphinxAtStartPar
{\hyperref[\detokenize{ipynb/chapter1::doc}]{\sphinxcrossref{Protein parameters analysis}}}

\item {} 
\sphinxAtStartPar
{\hyperref[\detokenize{ipynb/chapter2::doc}]{\sphinxcrossref{Potential sites of chemical modification}}}

\item {} 
\sphinxAtStartPar
{\hyperref[\detokenize{ipynb/chapter3::doc}]{\sphinxcrossref{Structural analysis}}}

\item {} 
\sphinxAtStartPar
{\hyperref[\detokenize{ipynb/chapter4::doc}]{\sphinxcrossref{Immunogenicity analysis}}}

\end{itemize}

\sphinxstepscope


\chapter{Protein parameters analysis}
\label{\detokenize{ipynb/chapter1:protein-parameters-analysis}}\label{\detokenize{ipynb/chapter1::doc}}
\sphinxAtStartPar
The program performs most of the same functions as the Expasy ProtParam tool.
Protein parameters for analysing protein sequences.

\sphinxAtStartPar
The program calculates:
\begin{enumerate}
\sphinxsetlistlabels{\arabic}{enumi}{enumii}{}{.}%
\item {} 
\sphinxAtStartPar
Molecular weight

\item {} 
\sphinxAtStartPar
Chemicalcomposition

\item {} 
\sphinxAtStartPar
Amino acid composition

\item {} 
\sphinxAtStartPar
pI

\item {} 
\sphinxAtStartPar
Extinction coefficient

\item {} 
\sphinxAtStartPar
Aromaticity

\item {} 
\sphinxAtStartPar
GRAVY and Instability\_index

\end{enumerate}


\section{Input sequences}
\label{\detokenize{ipynb/chapter1:input-sequences}}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{SDTGRPFVEM} \PYG{n}{YSEIPEIIHM} \PYG{n}{TEGRELVIPC} \PYG{n}{RVTSPNITVT} \PYG{n}{LKKFPLDTLI} \PYG{n}{PDGKRIIWDS}
\PYG{n}{RKGFIISNAT} \PYG{n}{YKEIGLLTCE} \PYG{n}{ATVNGHLYKT} \PYG{n}{NYLTHRQTNT} \PYG{n}{IIDVVLSPSH} \PYG{n}{GIELSVGEKL}
\PYG{n}{VLNCTARTEL} \PYG{n}{NVGIDFNWEY} \PYG{n}{PSSKHQHKKL} \PYG{n}{VNRDLKTQSG} \PYG{n}{SEMKKFLSTL} \PYG{n}{TIDGVTRSDQ}
\PYG{n}{GLYTCAASSG} \PYG{n}{LMTKKNSTFV} \PYG{n}{RVHEKDKTHT} \PYG{n}{CPPCPAPELL} \PYG{n}{GGPSVFLFPP} \PYG{n}{KPKDTLMISR}
\PYG{n}{TPEVTCVVVD} \PYG{n}{VSHEDPEVKF} \PYG{n}{NWYVDGVEVH} \PYG{n}{NAKTKPREEQ} \PYG{n}{YNSTYRVVSV} \PYG{n}{LTVLHQDWLN}
\PYG{n}{GKEYKCKVSN} \PYG{n}{KALPAPIEKT} \PYG{n}{ISKAKGQPRE} \PYG{n}{PQVYTLPPSR} \PYG{n}{DELTKNQVSL} \PYG{n}{TCLVKGFYPS}
\PYG{n}{DIAVEWESNG} \PYG{n}{QPENNYKTTP} \PYG{n}{PVLDSDGSFF} \PYG{n}{LYSKLTVDKS} \PYG{n}{RWQQGNVFSC} \PYG{n}{SVMHEALHNH}
\PYG{n}{YTQKSLSLSP} \PYG{n}{G}
\end{sphinxVerbatim}

\sphinxAtStartPar
(Known Disulfide bridge: 30\sphinxhyphen{}79, 124\sphinxhyphen{}185, 246\sphinxhyphen{}306, 352\sphinxhyphen{}410, 211\sphinxhyphen{}211’, 214\sphinxhyphen{}214’)


\section{Molecular weight}
\label{\detokenize{ipynb/chapter1:molecular-weight}}
\sphinxAtStartPar
Amino acids are the building blocks that form polypeptides and ultimately proteins. Calculates the molecular weight of a protein.

\begin{sphinxuseclass}{cell}
\begin{sphinxuseclass}{tag_remove-input}\begin{sphinxVerbatimOutput}

\begin{sphinxuseclass}{cell_output}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
Molecular weigh of Aflibercept is 48,458.67 Da.
Total number of amino acid in Aflibercept is 431.
\end{sphinxVerbatim}

\end{sphinxuseclass}\end{sphinxVerbatimOutput}

\end{sphinxuseclass}
\end{sphinxuseclass}

\section{Chemical composition}
\label{\detokenize{ipynb/chapter1:chemical-composition}}
\sphinxAtStartPar
A chemical formula is a way of presenting information about the chemical proportions of atoms that constitute a particular chemical compound or molecule, using chemical element symbols, numbers, and sometimes also other symbols, such as parentheses, dashes, brackets, commas and plus (+) and minus (−) signs. These are limited to a single typographic line of symbols, which may include subscripts and superscripts.

\begin{sphinxuseclass}{cell}
\begin{sphinxuseclass}{tag_remove-input}\begin{sphinxVerbatimOutput}

\begin{sphinxuseclass}{cell_output}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
Chemical formula of Aflibercept is C2159H3404N582O652S16.
Total number of atom in Aflibercept is 6813.
\end{sphinxVerbatim}

\end{sphinxuseclass}\end{sphinxVerbatimOutput}

\end{sphinxuseclass}
\end{sphinxuseclass}

\section{Amino acid composition}
\label{\detokenize{ipynb/chapter1:amino-acid-composition}}
\sphinxAtStartPar
We can easily count the number of each type of amino acid.


\subsection{Number of each amino acids}
\label{\detokenize{ipynb/chapter1:number-of-each-amino-acids}}
\sphinxAtStartPar
Simply counts the number times an amino acid is repeated in the protein sequence.

\begin{sphinxuseclass}{cell}
\begin{sphinxuseclass}{tag_remove-input}\begin{sphinxVerbatimOutput}

\begin{sphinxuseclass}{cell_output}
\noindent\sphinxincludegraphics{{chapter1_5_0}.png}

\end{sphinxuseclass}\end{sphinxVerbatimOutput}

\end{sphinxuseclass}
\end{sphinxuseclass}
\begin{sphinxuseclass}{cell}
\begin{sphinxuseclass}{tag_remove-input}\begin{sphinxVerbatimOutput}

\begin{sphinxuseclass}{cell_output}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
Total number of positively charged residues (Arg + Lys) of Aflibercept is 51.
Total number of negatively charged residues (Asp + Glu) of Aflibercept is 48.
\end{sphinxVerbatim}

\end{sphinxuseclass}\end{sphinxVerbatimOutput}

\end{sphinxuseclass}
\end{sphinxuseclass}

\subsection{Percent of amino acid contents}
\label{\detokenize{ipynb/chapter1:percent-of-amino-acid-contents}}
\sphinxAtStartPar
The same as number of amino acid, only returns the number in percentage of entire sequence. When the total is taken as 100, the ratio of each amino acid is calculated.

\begin{sphinxuseclass}{cell}
\begin{sphinxuseclass}{tag_remove-input}\begin{sphinxVerbatimOutput}

\begin{sphinxuseclass}{cell_output}
\noindent\sphinxincludegraphics{{chapter1_8_0}.png}

\end{sphinxuseclass}\end{sphinxVerbatimOutput}

\end{sphinxuseclass}
\end{sphinxuseclass}

\section{Theoretical pI}
\label{\detokenize{ipynb/chapter1:theoretical-pi}}
\sphinxAtStartPar
The isoelectric point (pI, pH(I), IEP), is the pH at which a molecule carries no net electrical charge or is electrically neutral in the statistical mean.

\sphinxAtStartPar
The pI value can affect the solubility of a molecule at a given pH. Such molecules have minimum solubility in water or salt solutions at the pH that corresponds to their pI and often precipitate out of solution. Biological amphoteric molecules such as proteins contain both acidic and basic functional groups.

\begin{sphinxuseclass}{cell}
\begin{sphinxuseclass}{tag_remove-input}\begin{sphinxVerbatimOutput}

\begin{sphinxuseclass}{cell_output}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
Theoretical pI is 8.198.
\end{sphinxVerbatim}

\end{sphinxuseclass}\end{sphinxVerbatimOutput}

\end{sphinxuseclass}
\end{sphinxuseclass}

\subsection{Charge of target protein ins PBS}
\label{\detokenize{ipynb/chapter1:charge-of-target-protein-ins-pbs}}
\sphinxAtStartPar
Phosphate\sphinxhyphen{}buffered saline (abbreviated PBS) is a buffer solution (pH \textasciitilde{} 7.4) commonly used in biological research.

\begin{sphinxadmonition}{note}{Note:}
\sphinxAtStartPar
PBS has many uses because it is isotonic and non\sphinxhyphen{}toxic to most cells
\end{sphinxadmonition}

\begin{sphinxuseclass}{cell}
\begin{sphinxuseclass}{tag_remove-input}\begin{sphinxVerbatimOutput}

\begin{sphinxuseclass}{cell_output}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
Aflibercept is positive charged(+2.436) in PBS
\end{sphinxVerbatim}

\end{sphinxuseclass}\end{sphinxVerbatimOutput}

\end{sphinxuseclass}
\end{sphinxuseclass}

\section{Extinction coefficient}
\label{\detokenize{ipynb/chapter1:extinction-coefficient}}
\sphinxAtStartPar
Extinction (or extinction coefficient) is defined as the ratio of maximum to minimum transmission of a beam of light that passes through a polarization optical train.

\sphinxAtStartPar
extinction coefficient in units of  M\sphinxhyphen{}1 cm\sphinxhyphen{}1, at 280 nm measured in water.


\subsection{with reduced cysteines}
\label{\detokenize{ipynb/chapter1:with-reduced-cysteines}}
\begin{sphinxuseclass}{cell}
\begin{sphinxuseclass}{tag_remove-input}\begin{sphinxVerbatimOutput}

\begin{sphinxuseclass}{cell_output}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
Extinction coefficient of Aflibercept at reduced condition is 55350.
Abs 0.1\PYGZpc{}(=1 g/L) is 1.142.
\end{sphinxVerbatim}

\end{sphinxuseclass}\end{sphinxVerbatimOutput}

\end{sphinxuseclass}
\end{sphinxuseclass}

\subsection{with non\sphinxhyphen{}reduced cysteines}
\label{\detokenize{ipynb/chapter1:with-non-reduced-cysteines}}
\begin{sphinxuseclass}{cell}
\begin{sphinxuseclass}{tag_remove-input}\begin{sphinxVerbatimOutput}

\begin{sphinxuseclass}{cell_output}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
Extinction coefficient of Aflibercept at non\PYGZhy{}reduced condition is 55975.
Abs 0.1\PYGZpc{}(=1 g/L) is 1.155.
\end{sphinxVerbatim}

\end{sphinxuseclass}\end{sphinxVerbatimOutput}

\end{sphinxuseclass}
\end{sphinxuseclass}

\section{Aromaticity}
\label{\detokenize{ipynb/chapter1:aromaticity}}
\sphinxAtStartPar
Calculate the aromaticity according to Lobry, 1994. Calculates the aromaticity value of a protein according to Lobry, 1994. It is simply the relative frequency of Phe+Trp+Tyr.

\begin{sphinxuseclass}{cell}
\begin{sphinxuseclass}{tag_remove-input}\begin{sphinxVerbatimOutput}

\begin{sphinxuseclass}{cell_output}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
The Aromaticity of target protein is 7.889\PYGZpc{}.
\end{sphinxVerbatim}

\end{sphinxuseclass}\end{sphinxVerbatimOutput}

\end{sphinxuseclass}
\end{sphinxuseclass}

\section{GRAVY}
\label{\detokenize{ipynb/chapter1:gravy}}
\sphinxAtStartPar
Protein GRAVY returns the GRAVY (grand average of hydropathy) value for the protein sequences you enter. The GRAVY value is calculated by adding the hydropathy value for each residue and dividing by the length of the sequence (Kyte and Doolittle; 1982).

\sphinxAtStartPar
A higher value is more hydrophobic. A lower value is more hydrophilic.

\begin{sphinxuseclass}{cell}
\begin{sphinxuseclass}{tag_remove-input}\begin{sphinxVerbatimOutput}

\begin{sphinxuseclass}{cell_output}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
Aflibercept is kind a more hydrophilic protein. The GRAVY value is \PYGZhy{}0.460.
\end{sphinxVerbatim}

\end{sphinxuseclass}\end{sphinxVerbatimOutput}

\end{sphinxuseclass}
\end{sphinxuseclass}

\section{Instability\_index}
\label{\detokenize{ipynb/chapter1:instability-index}}
\sphinxAtStartPar
Implementation of the method of Guruprasad et al. (1990, Protein Engineering, 4, 155\sphinxhyphen{}161). This method tests a protein for stability. Any value above 40 means the protein is unstable (=has a short half life).

\begin{sphinxuseclass}{cell}
\begin{sphinxuseclass}{tag_remove-input}\begin{sphinxVerbatimOutput}

\begin{sphinxuseclass}{cell_output}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
The instability index of Aflibercept is computed to be 39.222, and it seems stable.
\end{sphinxVerbatim}

\end{sphinxuseclass}\end{sphinxVerbatimOutput}

\end{sphinxuseclass}
\end{sphinxuseclass}
\sphinxstepscope


\chapter{Potential sites of chemical modification}
\label{\detokenize{ipynb/chapter2:potential-sites-of-chemical-modification}}\label{\detokenize{ipynb/chapter2::doc}}
\sphinxAtStartPar
An initial scan of the antibody sequences is presented based purely upon sequence. If a structural analysis was also requested, this section should be used in conjunction with the molecular surface analysis described in a subsequent section. Any of the sites listed below could be candidates for further consideration if the molecular surface analysis shows that they are significantly exposed on the surface of the antibody, increasing their propensity for chemical modification. The canonical sequence analysis is also helpful here, since each of these sites can also be considered in the context of their frequency of occurrence within the canonical library of homologous sequences.

\sphinxAtStartPar
This is an example of a footnote.%
\begin{footnote}[1]\sphinxAtStartFootnote
The definition for referencing footnotes is generally placed at the bottom of the document.
%
\end{footnote}

\begin{sphinxuseclass}{cell}
\begin{sphinxuseclass}{tag_remove-input}
\end{sphinxuseclass}
\end{sphinxuseclass}

\section{Potential deamidation positions}
\label{\detokenize{ipynb/chapter2:potential-deamidation-positions}}
\sphinxAtStartPar
Asparagine (N) and glutamine (Q) residues are particularly prone to deamidation when they are followed in the sequence by amino acids with smaller side chains, that leave the intervening peptide group more exposed. Deamidation proceeds much more quickly if the susceptible amino acid is followed by a small, flexible residue such as glycine whose low steric hindrance leaves the peptide group open for attack.


\subsection{Search patterns: ASN/GLN\sphinxhyphen{}ALA/GLY/SER/THR}
\label{\detokenize{ipynb/chapter2:search-patterns-asn-gln-ala-gly-ser-thr}}
\begin{sphinxuseclass}{cell}
\begin{sphinxuseclass}{tag_remove-input}\begin{sphinxVerbatimOutput}

\begin{sphinxuseclass}{cell_output}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
68\PYGZhy{}NA\PYGZhy{}69
84\PYGZhy{}NG\PYGZhy{}85
97\PYGZhy{}QT\PYGZhy{}98
99\PYGZhy{}NT\PYGZhy{}100
158\PYGZhy{}QS\PYGZhy{}159
180\PYGZhy{}QG\PYGZhy{}181
196\PYGZhy{}NS\PYGZhy{}197
271\PYGZhy{}NA\PYGZhy{}272
282\PYGZhy{}NS\PYGZhy{}283
300\PYGZhy{}NG\PYGZhy{}301
369\PYGZhy{}NG\PYGZhy{}370
404\PYGZhy{}QG\PYGZhy{}405
\end{sphinxVerbatim}

\end{sphinxuseclass}\end{sphinxVerbatimOutput}

\end{sphinxuseclass}
\end{sphinxuseclass}

\section{Potential o\sphinxhyphen{}linked glycosylation sites}
\label{\detokenize{ipynb/chapter2:potential-o-linked-glycosylation-sites}}
\sphinxAtStartPar
The O\sphinxhyphen{}linked glycosylation of serine and threonine residues seems to be particularly sensitive to
the presence of one or more proline residues in their vicinity in the sequence, particularly in the2
\sphinxhyphen{}1 and +3 positions.


\subsection{Search patterns: PRO\sphinxhyphen{}SER/THR}
\label{\detokenize{ipynb/chapter2:search-patterns-pro-ser-thr}}
\begin{sphinxuseclass}{cell}
\begin{sphinxuseclass}{tag_remove-input}\begin{sphinxVerbatimOutput}

\begin{sphinxuseclass}{cell_output}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
108\PYGZhy{}PS\PYGZhy{}109
141\PYGZhy{}PS\PYGZhy{}142
223\PYGZhy{}PS\PYGZhy{}224
338\PYGZhy{}PS\PYGZhy{}339
359\PYGZhy{}PS\PYGZhy{}360
\end{sphinxVerbatim}

\end{sphinxuseclass}\end{sphinxVerbatimOutput}

\end{sphinxuseclass}
\end{sphinxuseclass}

\subsection{Search patterns: SER/THR\sphinxhyphen{}X\sphinxhyphen{}X\sphinxhyphen{}PRO}
\label{\detokenize{ipynb/chapter2:search-patterns-ser-thr-x-x-pro}}
\begin{sphinxuseclass}{cell}
\begin{sphinxuseclass}{tag_remove-input}\begin{sphinxVerbatimOutput}

\begin{sphinxuseclass}{cell_output}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
3\PYGZhy{}TGRP\PYGZhy{}6
12\PYGZhy{}SEIP\PYGZhy{}15
48\PYGZhy{}TLIP\PYGZhy{}51
210\PYGZhy{}TCPP\PYGZhy{}213
239\PYGZhy{}SRTP\PYGZhy{}242
335\PYGZhy{}TLPP\PYGZhy{}338
378\PYGZhy{}TTPP\PYGZhy{}381
427\PYGZhy{}SLSP\PYGZhy{}430
\end{sphinxVerbatim}

\end{sphinxuseclass}\end{sphinxVerbatimOutput}

\end{sphinxuseclass}
\end{sphinxuseclass}

\section{Potential n\sphinxhyphen{}linked glycosylation sites}
\label{\detokenize{ipynb/chapter2:potential-n-linked-glycosylation-sites}}

\subsection{Search patterns: ASN\sphinxhyphen{}X\sphinxhyphen{}SER/THR}
\label{\detokenize{ipynb/chapter2:search-patterns-asn-x-ser-thr}}
\begin{sphinxuseclass}{cell}
\begin{sphinxuseclass}{tag_remove-input}\begin{sphinxVerbatimOutput}

\begin{sphinxuseclass}{cell_output}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
36\PYGZhy{}NIT\PYGZhy{}38
68\PYGZhy{}NAT\PYGZhy{}70
123\PYGZhy{}NCT\PYGZhy{}125
196\PYGZhy{}NST\PYGZhy{}198
282\PYGZhy{}NST\PYGZhy{}284
\end{sphinxVerbatim}

\end{sphinxuseclass}\end{sphinxVerbatimOutput}

\end{sphinxuseclass}
\end{sphinxuseclass}

\chapter{Secondary structure fraction}
\label{\detokenize{ipynb/chapter2:secondary-structure-fraction}}
\sphinxAtStartPar
A very useful method — .secondary\_structure\_fraction() — returns the fraction of amino acids that tend to be found in the three classical secondary structures. These are beta sheets, alpha helixes, and turns (where the residues change direction).

\sphinxAtStartPar
Secondary\_structure\_fraction: This methods returns a list of the fraction of amino acids which tend to be in helix, turn or sheet.
\begin{itemize}
\item {} 
\sphinxAtStartPar
Amino acids in helix: V, I, Y, F, W, L.

\item {} 
\sphinxAtStartPar
Amino acids in turn: N, P, G, S.

\item {} 
\sphinxAtStartPar
Amino acids in sheet: E, M, A, L.

\end{itemize}

\begin{sphinxuseclass}{cell}
\begin{sphinxuseclass}{tag_remove-input}\begin{sphinxVerbatimOutput}

\begin{sphinxuseclass}{cell_output}
\noindent\sphinxincludegraphics{{chapter2_11_0}.png}

\end{sphinxuseclass}\end{sphinxVerbatimOutput}

\end{sphinxuseclass}
\end{sphinxuseclass}

\section{Secondary structure prediction}
\label{\detokenize{ipynb/chapter2:secondary-structure-prediction}}
\sphinxAtStartPar
Protein secondary structure prediction is one of the most important and challenging problems in bioinformatics. Here in, the P\sphinxhyphen{}SEA algorithm that to predict the secondary structures of proteins sequences based only on knowledge of their primary structure.

\begin{sphinxuseclass}{cell}
\begin{sphinxuseclass}{tag_remove-input}\begin{sphinxVerbatimOutput}

\begin{sphinxuseclass}{cell_output}
\noindent\sphinxincludegraphics{{chapter2_13_0}.png}

\end{sphinxuseclass}\end{sphinxVerbatimOutput}

\end{sphinxuseclass}
\end{sphinxuseclass}

\chapter{Protein Scales}
\label{\detokenize{ipynb/chapter2:protein-scales}}
\sphinxAtStartPar
Protein scales are a way of measuring certain attributes of residues over the length of the peptide sequence using a sliding window. Scales are comprised of values for each amino acid based on different physical and chemical properties, such as hydrophobicity, secondary structure tendencies, and surface accessibility. As opposed to some chain\sphinxhyphen{}level measures like overall molecule behavior, scales allow a more granular understanding of how smaller sections of the sequence will behave.

\sphinxAtStartPar
Some common scales include:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZhy{} kd → Kyte \PYGZam{} Doolittle Index of Hydrophobicity
\PYGZhy{} Flex → Normalized average flexibility parameters (B\PYGZhy{}values)
\PYGZhy{} hw → Hopp \PYGZam{} Wood Index of Hydrophilicity
\PYGZhy{} em → Emini Surface fractional probability (Surface Accessibility)
\end{sphinxVerbatim}


\section{Hydrophobicity index}
\label{\detokenize{ipynb/chapter2:hydrophobicity-index}}
\sphinxAtStartPar
hydrophobicity is the physical property of a molecule that is seemingly repelled from a mass of water (known as a hydrophobe).

\begin{sphinxuseclass}{cell}
\begin{sphinxuseclass}{tag_remove-input}\begin{sphinxVerbatimOutput}

\begin{sphinxuseclass}{cell_output}
\noindent\sphinxincludegraphics{{chapter2_15_0}.png}

\end{sphinxuseclass}\end{sphinxVerbatimOutput}

\end{sphinxuseclass}
\end{sphinxuseclass}

\section{Hydrophilicity index}
\label{\detokenize{ipynb/chapter2:hydrophilicity-index}}
\sphinxAtStartPar
Hydrophilicity is the tendency of a molecule to be solvated by water.

\begin{sphinxuseclass}{cell}
\begin{sphinxuseclass}{tag_remove-input}\begin{sphinxVerbatimOutput}

\begin{sphinxuseclass}{cell_output}
\noindent\sphinxincludegraphics{{chapter2_17_0}.png}

\end{sphinxuseclass}\end{sphinxVerbatimOutput}

\end{sphinxuseclass}
\end{sphinxuseclass}

\section{Flexibility index}
\label{\detokenize{ipynb/chapter2:flexibility-index}}
\sphinxAtStartPar
Proteins are dynamic entities, and they possess an inherent flexibility that allows them to function through molecular interactions within the cell.

\begin{sphinxuseclass}{cell}
\begin{sphinxuseclass}{tag_remove-input}\begin{sphinxVerbatimOutput}

\begin{sphinxuseclass}{cell_output}
\noindent\sphinxincludegraphics{{chapter2_19_0}.png}

\end{sphinxuseclass}\end{sphinxVerbatimOutput}

\end{sphinxuseclass}
\end{sphinxuseclass}

\section{Surface accessibility}
\label{\detokenize{ipynb/chapter2:surface-accessibility}}
\sphinxAtStartPar
Data describing the solvent\sphinxhyphen{}accessible surface of a molecule is of great utility in the
development of that molecule as a therapeutic, particularly in the case of antibodies. In the
context of this report, the most obvious application of molecular surface data is in combination
with the potential sites of chemical modification, described in the previous section. Antibodies
(and other proteins) are known to undergo many different chemical modifications as a result of
interactions with their aqueous environment. The probability and kinetic rate of such a
modification is greatly enhanced by the degree of exposure of the potential modification site to
the solvent environment.
If a 3D structure was supplied or requested by the client, then the graphs below will show the
solvent\sphinxhyphen{}accessible surface areas in (Å2) calculated for each residue in the light and heavy chain
variable regions of the antibody. These solvent\sphinxhyphen{}accessible surfaces are calculated using the
“rolling ball” algorithm developed by Shrake \& Rupley in which a spherical solvent “probe” is3
used, the accessible surface being generated by tracing the center of the probe sphere as it
rolls along the van der Waals surface (as shown below).

\sphinxAtStartPar
The solvent\sphinxhyphen{}accessible surface for each residue depends upon the degree of exposure of the
residue on the antibody surface, but also on the size of the residue side chain. Residues that
are completely buried within the hydrophobic core of the antibody, will have solvent\sphinxhyphen{}accessible
surface areas at or close to zero. Residues that are highly exposed at the antibody surface will
have values that range from around 100 \sphinxhyphen{} 200 Å2, depending upon the size of the residue side
chain and the degree of exposure.

\begin{sphinxuseclass}{cell}
\begin{sphinxuseclass}{tag_remove-input}\begin{sphinxVerbatimOutput}

\begin{sphinxuseclass}{cell_output}
\noindent\sphinxincludegraphics{{chapter2_21_0}.png}

\end{sphinxuseclass}\end{sphinxVerbatimOutput}

\end{sphinxuseclass}
\end{sphinxuseclass}

\section{Instability index}
\label{\detokenize{ipynb/chapter2:instability-index}}
\sphinxAtStartPar
The instability index provides an estimate of the stability of your protein in a test tube. Statistical analysis of 12 unstable and 32 stable proteins has revealed that there are certain dipeptides, the occurence of which is significantly different in the unstable proteins compared with those in the stable ones.

\begin{sphinxuseclass}{cell}
\begin{sphinxuseclass}{tag_remove-input}\begin{sphinxVerbatimOutput}

\begin{sphinxuseclass}{cell_output}
\noindent\sphinxincludegraphics{{chapter2_23_0}.png}

\end{sphinxuseclass}\end{sphinxVerbatimOutput}

\end{sphinxuseclass}
\end{sphinxuseclass}

\bigskip\hrule\bigskip


\sphinxstepscope


\chapter{Structural analysis}
\label{\detokenize{ipynb/chapter3:structural-analysis}}\label{\detokenize{ipynb/chapter3::doc}}

\section{Detection of disulfide bonds}
\label{\detokenize{ipynb/chapter3:detection-of-disulfide-bonds}}
\sphinxAtStartPar
This function detects disulfide bridges in protein structures. Then the detected disulfide bonds are visualized and added to the bonds attribute of the AtomArray.,

\sphinxAtStartPar
The employed criteria for disulfide bonds are quite simple in this case: the atoms of two cystein residues must be in a vicinity of Å and the dihedral angle of must be .

\begin{sphinxuseclass}{cell}
\begin{sphinxuseclass}{tag_remove-input}\begin{sphinxVerbatimOutput}

\begin{sphinxuseclass}{cell_output}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
    A     352  CYS SG     S       \PYGZhy{}21.681    6.749   19.495
    A     410  CYS SG     S       \PYGZhy{}21.802    8.706   18.752
\end{sphinxVerbatim}

\end{sphinxuseclass}\end{sphinxVerbatimOutput}

\end{sphinxuseclass}
\end{sphinxuseclass}
\sphinxAtStartPar
The found disulfide bonds are visualized with the help of Matplotlib: The amino acid sequence is written on the X\sphinxhyphen{}axis and the disulfide bonds are depicted by yellow semi\sphinxhyphen{}ellipses.

\begin{sphinxuseclass}{cell}
\begin{sphinxuseclass}{tag_remove-input}\begin{sphinxVerbatimOutput}

\begin{sphinxuseclass}{cell_output}
\noindent\sphinxincludegraphics{{chapter3_3_0}.png}

\end{sphinxuseclass}\end{sphinxVerbatimOutput}

\end{sphinxuseclass}
\end{sphinxuseclass}

\section{Calculation of protein diameter}
\label{\detokenize{ipynb/chapter3:calculation-of-protein-diameter}}
\sphinxAtStartPar
This calculates the diameter of a protein defined as the maximum pairwise atom distance.

\begin{sphinxuseclass}{cell}
\begin{sphinxuseclass}{tag_remove-input}\begin{sphinxVerbatimOutput}

\begin{sphinxuseclass}{cell_output}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
Diameter of Aflibercept is: 145.705 Angstrom.
\end{sphinxVerbatim}

\end{sphinxuseclass}\end{sphinxVerbatimOutput}

\end{sphinxuseclass}
\end{sphinxuseclass}

\section{Ramachandran plot}
\label{\detokenize{ipynb/chapter3:ramachandran-plot}}
\sphinxAtStartPar
This creates a Ramachandran plot of the motor domain of target protein.

\begin{sphinxuseclass}{cell}
\begin{sphinxuseclass}{tag_remove-input}\begin{sphinxVerbatimOutput}

\begin{sphinxuseclass}{cell_output}
\noindent\sphinxincludegraphics{{chapter3_7_0}.png}

\end{sphinxuseclass}\end{sphinxVerbatimOutput}

\end{sphinxuseclass}
\end{sphinxuseclass}
\sphinxstepscope


\chapter{Immunogenicity analysis}
\label{\detokenize{ipynb/chapter4:immunogenicity-analysis}}\label{\detokenize{ipynb/chapter4::doc}}
\sphinxAtStartPar
We uses the method of removing and/or reducing potential T\sphinxhyphen{}cell epitopes, as an approach to the management of the immunogenicity of biologics. The protein sequence is scanned in silico, for sequences that have a strong binding signature for a family of 50 MHC Class II receptors , whose alleles cover 96 – 98\% of the human population. The presented histograms for each variable region sequence, show the average (for the n positively\sphinxhyphen{}testing MHC II alleles) of epitope strength at each position as a percentage for all epitopes above a threshold of 20\%. At each position in the sequence, the number of alleles scoring above the threshold is shown above the histogram at that position. The epitopes of most concern for the antibody’s immunogenicity are therefore those that have not just the highest average score per allele (as shown by the histogram), but which also score above the threshold across more alleles, since these epitopes are more likely to engender an immune response in a larger fraction of the patient population.

\sphinxAtStartPar
Experience using in silico algorithms of this kind in conjunction with laboratory immunogenicity assays has shown that epitopes below this threshold do not generally contribute significantly to the protein’s immunogenicity. The number of alleles, the affected alleles and their individual scores are also listed in the detailed analyses below each histogram figure.

\sphinxAtStartPar
The raw immunogenicity score quoted is the total over all epitopes above the threshold for all affected alleles. The normalized immunogenicity score is this raw score divided by the sequence length, and represents epitope strength per unit sequence to enable comparisons of protein sequences of different lengths.

\sphinxAtStartPar
The absolute magnitudes of these scores are somewhat arbitrary, but they have value as comparative metrics. It has been shown that human serum proteins generally display an immunogenicity potential that is inversely proportional to their abundance in serum . Proteins that are found at very low concentrations in serum, like erythropoietin, can have normalized scores above 80\%. By contrast, very abundant human serum proteins like albumin and immunoglobulins typically have normalized scores in the 35 \sphinxhyphen{} 50\% range.

\begin{sphinxuseclass}{cell}
\begin{sphinxuseclass}{tag_remove-input}
\begin{sphinxuseclass}{tag_remove-output}
\end{sphinxuseclass}
\end{sphinxuseclass}
\end{sphinxuseclass}

\section{MHC class 1}
\label{\detokenize{ipynb/chapter4:mhc-class-1}}
\sphinxAtStartPar
Class I major histocompatibility complex (MHC) molecules bind, and present to T cells, short peptides derived from intracellular processing of proteins. The peptide repertoire of a specific molecule is to a large extent determined by the molecular structure accommodating so\sphinxhyphen{}called main anchor positions of the presented peptide.

\sphinxAtStartPar
MHC class I molecules are one of two primary classes of major histocompatibility complex (MHC) molecules (the other being MHC class II) and are found on the cell surface of all nucleated cells in the bodies of vertebrates.

\sphinxAtStartPar
Their function is to display peptide fragments of proteins from within the cell to cytotoxic T cells; this will trigger an immediate response from the immune system against a particular non\sphinxhyphen{}self antigen displayed with the help of an MHC class I protein. Because MHC class I molecules present peptides derived from cytosolic proteins, the pathway of MHC class I presentation is often called cytosolic or endogenous pathway.%
\begin{footnote}[1]\sphinxAtStartFootnote
Kimball’s Biology Pages, Histocompatibility Molecules
%
\end{footnote}

\sphinxAtStartPar
In humans, the HLAs corresponding to MHC class I are HLA\sphinxhyphen{}A, HLA\sphinxhyphen{}B, and HLA\sphinxhyphen{}C. Almost a decade ago we introduced this concept of clustering human leukocyte antigen (HLA) alleles and defined nine different groups, denominated as supertypes, on the basis of their main anchor specificity.
\begin{itemize}
\item {} 
\sphinxAtStartPar
HLA\sphinxhyphen{}A01:01, HLA\sphinxhyphen{}A02:01, HLA\sphinxhyphen{}A03:01, HLA\sphinxhyphen{}A24:02, HLA\sphinxhyphen{}B07:02, HLA\sphinxhyphen{}B40:01

\end{itemize}

\begin{sphinxuseclass}{cell}
\begin{sphinxuseclass}{tag_remove-input}
\begin{sphinxuseclass}{tag_remove-output}
\end{sphinxuseclass}
\end{sphinxuseclass}
\end{sphinxuseclass}
\begin{sphinxuseclass}{cell}
\begin{sphinxuseclass}{tag_remove-input}\begin{sphinxVerbatimOutput}

\begin{sphinxuseclass}{cell_output}
\noindent\sphinxincludegraphics{{chapter4_4_0}.png}

\end{sphinxuseclass}\end{sphinxVerbatimOutput}

\end{sphinxuseclass}
\end{sphinxuseclass}

\subsection{MHC class1 binding peptide}
\label{\detokenize{ipynb/chapter4:mhc-class1-binding-peptide}}

\subsubsection{Top10 strong binding peptide}
\label{\detokenize{ipynb/chapter4:top10-strong-binding-peptide}}
\begin{sphinxuseclass}{cell}
\begin{sphinxuseclass}{tag_remove-input}\begin{sphinxVerbatimOutput}

\begin{sphinxuseclass}{cell_output}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
        allele      peptide       Core   Rank
0  HLA\PYGZhy{}A*01:01    DSDGSFFLY  DSDGSFFLY  0.004
1  HLA\PYGZhy{}A*03:01    ALHNHYTQK  ALHNHYTQK  0.013
2  HLA\PYGZhy{}B*07:02   QPREPQVYTL  QPREPQVTL  0.013
3  HLA\PYGZhy{}A*01:01   LDSDGSFFLY  LSDGSFFLY  0.027
4  HLA\PYGZhy{}A*01:01  VLDSDGSFFLY  VSDGSFFLY  0.028
5  HLA\PYGZhy{}B*40:01    SEIPEIIHM  SEIPEIIHM  0.030
6  HLA\PYGZhy{}A*03:01    ATVNGHLYK  ATVNGHLYK  0.030
7  HLA\PYGZhy{}A*02:01    LMISRTPEV  LMISRTPEV  0.031
8  HLA\PYGZhy{}B*40:01    IELSVGEKL  IELSVGEKL  0.032
9  HLA\PYGZhy{}A*03:01    STFVRVHEK  STFVRVHEK  0.038
\end{sphinxVerbatim}

\end{sphinxuseclass}\end{sphinxVerbatimOutput}

\end{sphinxuseclass}
\end{sphinxuseclass}

\subsubsection{Frequency of binding peptide}
\label{\detokenize{ipynb/chapter4:frequency-of-binding-peptide}}
\sphinxAtStartPar
Pie graphs are drawn to visualize peptides with immunogenicity in various alleles.

\begin{sphinxuseclass}{cell}
\begin{sphinxuseclass}{tag_remove-input}\begin{sphinxVerbatimOutput}

\begin{sphinxuseclass}{cell_output}
\noindent\sphinxincludegraphics{{chapter4_8_0}.png}

\end{sphinxuseclass}\end{sphinxVerbatimOutput}

\end{sphinxuseclass}
\end{sphinxuseclass}

\section{MHC class 2}
\label{\detokenize{ipynb/chapter4:mhc-class-2}}
\begin{sphinxuseclass}{cell}
\begin{sphinxuseclass}{tag_remove-input}
\begin{sphinxuseclass}{tag_remove-output}
\end{sphinxuseclass}
\end{sphinxuseclass}
\end{sphinxuseclass}
\begin{sphinxuseclass}{cell}
\begin{sphinxuseclass}{tag_remove-input}\begin{sphinxVerbatimOutput}

\begin{sphinxuseclass}{cell_output}
\noindent\sphinxincludegraphics{{chapter4_12_0}.png}

\end{sphinxuseclass}\end{sphinxVerbatimOutput}

\end{sphinxuseclass}
\end{sphinxuseclass}

\subsection{MHC class 2 binder peptide}
\label{\detokenize{ipynb/chapter4:mhc-class-2-binder-peptide}}

\subsubsection{Top10 peptide}
\label{\detokenize{ipynb/chapter4:top10-peptide}}
\begin{sphinxuseclass}{cell}
\begin{sphinxuseclass}{tag_remove-input}\begin{sphinxVerbatimOutput}

\begin{sphinxuseclass}{cell_output}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
      allele          peptide       Core  Rank
0  DRB1\PYGZus{}0103  GKEYKCKVSNKALPA  YKCKVSNKA  0.06
1  DRB1\PYGZus{}0103  NGKEYKCKVSNKALP  YKCKVSNKA  0.09
2  DRB1\PYGZus{}0103  KEYKCKVSNKALPAP  YKCKVSNKA  0.25
3  DRB1\PYGZus{}0103  LNGKEYKCKVSNKAL  YKCKVSNKA  0.33
4  DRB1\PYGZus{}0106  PEIIHMTEGRELVIP  IHMTEGREL  0.36
5  DRB1\PYGZus{}0109  GKEYKCKVSNKALPA  YKCKVSNKA  0.49
6  DRB1\PYGZus{}0104  PEIIHMTEGRELVIP  IHMTEGREL  0.52
7  DRB1\PYGZus{}0103  RIIWDSRKGFIISNA  WDSRKGFII  0.57
8  DRB1\PYGZus{}0102  PEIIHMTEGRELVIP  IHMTEGREL  0.61
9  DRB1\PYGZus{}0109  NGKEYKCKVSNKALP  YKCKVSNKA  0.74
\end{sphinxVerbatim}

\end{sphinxuseclass}\end{sphinxVerbatimOutput}

\end{sphinxuseclass}
\end{sphinxuseclass}

\subsubsection{Frequency of binding peptide}
\label{\detokenize{ipynb/chapter4:id2}}
\sphinxAtStartPar
Pie graphs are drawn to visualize peptides with immunogenicity in various alleles.

\begin{sphinxuseclass}{cell}
\begin{sphinxuseclass}{tag_remove-input}\begin{sphinxVerbatimOutput}

\begin{sphinxuseclass}{cell_output}
\noindent\sphinxincludegraphics{{chapter4_16_0}.png}

\end{sphinxuseclass}\end{sphinxVerbatimOutput}

\end{sphinxuseclass}
\end{sphinxuseclass}

\chapter{Appendix}
\label{\detokenize{ipynb/chapter4:appendix}}
\begin{sphinxuseclass}{cell}
\begin{sphinxuseclass}{tag_remove-input}
\begin{sphinxuseclass}{tag_remove-output}
\end{sphinxuseclass}
\end{sphinxuseclass}
\end{sphinxuseclass}
\begin{sphinxuseclass}{cell}
\begin{sphinxuseclass}{tag_remove-input}
\begin{sphinxuseclass}{tag_remove-output}
\end{sphinxuseclass}
\end{sphinxuseclass}
\end{sphinxuseclass}
\begin{sphinxuseclass}{cell}
\begin{sphinxuseclass}{tag_remove-input}
\begin{sphinxuseclass}{tag_remove-output}
\end{sphinxuseclass}
\end{sphinxuseclass}
\end{sphinxuseclass}
\begin{sphinxuseclass}{cell}
\begin{sphinxuseclass}{tag_remove-input}
\begin{sphinxuseclass}{tag_remove-output}
\end{sphinxuseclass}
\end{sphinxuseclass}
\end{sphinxuseclass}

\bigskip\hrule\bigskip








\renewcommand{\indexname}{Index}
\printindex
\end{document}